{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fd650268",
   "metadata": {},
   "source": [
    "# Phylogenetic analysis\n",
    "\n",
    "For phylogenetic analysis we use the file with amino acid FASTA sequences generated in the folder `OrthoGroups_analysis`.\n",
    "\n",
    "Versions:\n",
    "* MUSCLE - v3.8.1551\n",
    "* MAFFT - v7.310\n",
    "* ClustalO - 1.2.3\n",
    "* Ugene - v41\n",
    "* [IQ-Tree v2.2.0_beta](https://anaconda.org/bioconda/iqtree)\n",
    "\n",
    "Coverage is universally bad, but we decided to use ClustalO to build the tree."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c57063e5",
   "metadata": {},
   "source": [
    "## MUSCLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3cf915af",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "\r\n",
      "*** WARNING *** Invalid character '*' in FASTA sequence data, ignored\r\n",
      "conc_protein_seq 201 seqs, lengths min 43, max 286, avg 135\r\n",
      "00:00:00    15 MB(-1%)  Iter   1    0.00%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1    2.47%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1    4.93%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1    7.39%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1    9.86%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   12.32%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   14.78%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   17.25%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   19.71%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   22.17%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   24.63%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   27.10%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   29.56%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   32.02%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   34.49%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   36.95%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   39.41%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   41.87%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   44.34%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   46.80%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   49.26%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   51.73%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   54.19%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   56.65%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   59.12%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   61.58%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   64.04%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   66.50%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   68.97%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   71.43%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   73.89%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   76.36%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   78.82%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   81.28%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   83.74%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   86.21%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   88.67%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   91.13%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   93.60%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   96.06%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1   98.52%  K-mer dist pass 1\r",
      "00:00:00    15 MB(-1%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00    15 MB(-1%)  Iter   1    0.00%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1    2.47%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1    4.93%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1    7.39%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1    9.86%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   12.32%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   14.78%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   17.25%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   19.71%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   22.17%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   24.63%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   27.10%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   29.56%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   32.02%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   34.49%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   36.95%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   39.41%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   41.87%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   44.34%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   46.80%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   49.26%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   51.73%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   54.19%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   56.65%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   59.12%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   61.58%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   64.04%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   66.50%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   68.97%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   71.43%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   73.89%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   76.36%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   78.82%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   81.28%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   83.74%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   86.21%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   88.67%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   91.13%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   93.60%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   96.06%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1   98.52%  K-mer dist pass 2\r",
      "00:00:00    15 MB(-1%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00    16 MB(-1%)  Iter   1    0.50%  Align node       \r",
      "00:00:00    17 MB(-1%)  Iter   1    1.00%  Align node\r",
      "00:00:00    17 MB(-1%)  Iter   1    1.50%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    2.00%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    2.50%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    3.00%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    3.50%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    4.00%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    4.50%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    5.00%  Align node\r",
      "00:00:00    18 MB(-1%)  Iter   1    5.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    6.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    6.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    7.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    7.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    8.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    8.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    9.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1    9.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1   10.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1   10.50%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1   11.00%  Align node\r",
      "00:00:00    19 MB(-1%)  Iter   1   11.50%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   12.00%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   12.50%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   13.00%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   13.50%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   14.00%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   14.50%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   15.00%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   15.50%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   16.00%  Align node\r",
      "00:00:00    20 MB(-1%)  Iter   1   16.50%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   17.00%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   17.50%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   18.00%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   18.50%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   19.00%  Align node\r",
      "00:00:00    21 MB(-1%)  Iter   1   19.50%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   20.00%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   20.50%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   21.00%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   21.50%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   22.00%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   22.50%  Align node\r",
      "00:00:00    22 MB(-1%)  Iter   1   23.00%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   23.50%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   24.00%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   24.50%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   25.00%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   25.50%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   26.00%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   26.50%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   27.00%  Align node\r",
      "00:00:00    23 MB(-1%)  Iter   1   27.50%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   28.00%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   28.50%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   29.00%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   29.50%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   30.00%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   30.50%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   31.00%  Align node\r",
      "00:00:00    24 MB(-1%)  Iter   1   31.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   32.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   32.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   33.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   33.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   34.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   34.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   35.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   35.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   36.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   36.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   37.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   37.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   38.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   38.50%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   39.00%  Align node\r",
      "00:00:00    25 MB(-1%)  Iter   1   39.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   40.00%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   40.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   41.00%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   41.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   42.00%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   42.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   43.00%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   43.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   44.00%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   44.50%  Align node\r",
      "00:00:00    26 MB(-1%)  Iter   1   45.00%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   45.50%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   46.00%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   46.50%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   47.00%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   47.50%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   48.00%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   48.50%  Align node\r",
      "00:00:00    27 MB(-1%)  Iter   1   49.00%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   49.50%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   50.00%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   50.50%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   51.00%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   51.50%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   52.00%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   52.50%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   53.00%  Align node\r",
      "00:00:00    28 MB(-2%)  Iter   1   53.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   54.00%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   54.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   55.00%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   55.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   56.00%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   56.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   57.00%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   57.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   58.00%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   58.50%  Align node\r",
      "00:00:00    29 MB(-2%)  Iter   1   59.00%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   59.50%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   60.00%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   60.50%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   61.00%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   61.50%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   62.00%  Align node\r",
      "00:00:00    30 MB(-2%)  Iter   1   62.50%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   63.00%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   63.50%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   64.00%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   64.50%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   65.00%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   65.50%  Align node\r",
      "00:00:00    31 MB(-2%)  Iter   1   66.00%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   66.50%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   67.00%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   67.50%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   68.00%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   68.50%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   69.00%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   69.50%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   70.00%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   70.50%  Align node\r",
      "00:00:00    32 MB(-2%)  Iter   1   71.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   71.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   72.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   72.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   73.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   73.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   74.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   74.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   75.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   75.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   76.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   76.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   77.00%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   77.50%  Align node\r",
      "00:00:00    33 MB(-2%)  Iter   1   78.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   78.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   79.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   79.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   80.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   80.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   81.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   81.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   82.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   82.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   83.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   83.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   84.00%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   84.50%  Align node\r",
      "00:00:00    34 MB(-2%)  Iter   1   85.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   85.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   86.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   86.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   87.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   87.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   88.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   88.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   89.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   89.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   90.00%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   90.50%  Align node\r",
      "00:00:00    35 MB(-2%)  Iter   1   91.00%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   91.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   92.00%  Align node\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00    36 MB(-2%)  Iter   1   92.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   93.00%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   93.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   94.00%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   94.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   95.00%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   95.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   96.00%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   96.50%  Align node\r",
      "00:00:00    36 MB(-2%)  Iter   1   97.00%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1   97.50%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1   98.00%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1   98.50%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1   99.00%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1   99.50%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1  100.00%  Align node\r",
      "00:00:00    37 MB(-2%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:00    37 MB(-2%)  Iter   1    0.50%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    1.00%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    1.49%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    1.99%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    2.49%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    2.99%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    3.48%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    3.98%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    4.48%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    4.98%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    5.47%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    5.97%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    6.47%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    6.97%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    7.46%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    7.96%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    8.46%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    8.96%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    9.45%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1    9.95%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   10.45%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   10.95%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   11.44%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   11.94%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   12.44%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   12.94%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   13.43%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   13.93%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   14.43%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   14.93%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   15.42%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   15.92%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   16.42%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   16.92%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   17.41%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   17.91%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   18.41%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   18.91%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   19.40%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   19.90%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   20.40%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   20.90%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   21.39%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   21.89%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   22.39%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   22.89%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   23.38%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   23.88%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   24.38%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   24.88%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   25.37%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   25.87%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   26.37%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   26.87%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   27.36%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   27.86%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   28.36%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   28.86%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   29.35%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   29.85%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   30.35%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   30.85%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   31.34%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   31.84%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   32.34%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   32.84%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   33.33%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   33.83%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   34.33%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   34.83%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   35.32%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   35.82%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   36.32%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   36.82%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   37.31%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   37.81%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   38.31%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   38.81%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   39.30%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   39.80%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   40.30%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   40.80%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   41.29%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   41.79%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   42.29%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   42.79%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   43.28%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   43.78%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   44.28%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   44.78%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   45.27%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   45.77%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   46.27%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   46.77%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   47.26%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   47.76%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   48.26%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   48.76%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   49.25%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   49.75%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   50.25%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   50.75%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   51.24%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   51.74%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   52.24%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   52.74%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   53.23%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   53.73%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   54.23%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   54.73%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   55.22%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   55.72%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   56.22%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   56.72%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   57.21%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   57.71%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   58.21%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   58.71%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   59.20%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   59.70%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   60.20%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   60.70%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   61.19%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   61.69%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   62.19%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   62.69%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   63.18%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   63.68%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   64.18%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   64.68%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   65.17%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   65.67%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   66.17%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   66.67%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   67.16%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   67.66%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   68.16%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   68.66%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   69.15%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   69.65%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   70.15%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   70.65%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   71.14%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   71.64%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   72.14%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   72.64%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   73.13%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   73.63%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   74.13%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   74.63%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   75.12%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   75.62%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   76.12%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   76.62%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   77.11%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   77.61%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   78.11%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   78.61%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   79.10%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   79.60%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   80.10%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   80.60%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   81.09%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   81.59%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   82.09%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   82.59%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   83.08%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   83.58%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   84.08%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   84.58%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   85.07%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   85.57%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   86.07%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   86.57%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   87.06%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   87.56%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   88.06%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   88.56%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   89.05%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   89.55%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   90.05%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   90.55%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   91.04%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   91.54%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   92.04%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   92.54%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   93.03%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   93.53%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   94.03%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   94.53%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   95.02%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   95.52%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   96.02%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   96.52%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   97.01%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   97.51%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   98.01%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   98.51%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   99.00%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1   99.50%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1  100.00%  Root alignment\r",
      "00:00:00    37 MB(-2%)  Iter   1  100.00%  Root alignment\r",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00    38 MB(-2%)  Iter   2    0.50%  Refine tree   \r",
      "00:00:00    38 MB(-2%)  Iter   2    1.01%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    1.51%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    2.01%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    2.51%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    3.02%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    3.52%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    4.02%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    4.52%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    5.03%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    5.53%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    6.03%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    6.53%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    7.04%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    7.54%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    8.04%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    8.54%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    9.05%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2    9.55%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   10.05%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   10.55%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   11.06%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   11.56%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   12.06%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   12.56%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   13.07%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   13.57%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   14.07%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   14.57%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   15.08%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   15.58%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   16.08%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   16.58%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   17.09%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   17.59%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   18.09%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   18.59%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   19.10%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   19.60%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   20.10%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   20.60%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   21.11%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   21.61%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   22.11%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   22.61%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   23.12%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   23.62%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   24.12%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   24.62%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   25.13%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   25.63%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   26.13%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   26.63%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   27.14%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   27.64%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   28.14%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   28.64%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   29.15%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   29.65%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   30.15%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   30.65%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   31.16%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   31.66%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   32.16%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   32.66%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   33.17%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   33.67%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   34.17%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   34.67%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   35.18%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   35.68%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   36.18%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   36.68%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   37.19%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   37.69%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   38.19%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   38.69%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   39.20%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   39.70%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   40.20%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   40.70%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   41.21%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   41.71%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   42.21%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   42.71%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   43.22%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   43.72%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   44.22%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   44.72%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   45.23%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   45.73%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   46.23%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   46.73%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   47.24%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   47.74%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   48.24%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   48.74%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   49.25%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   49.75%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   50.25%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   50.75%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   51.26%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   51.76%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   52.26%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   52.76%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   53.27%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   53.77%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   54.27%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   54.77%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   55.28%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   55.78%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   56.28%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   56.78%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   57.29%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   57.79%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   58.29%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   58.79%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   59.30%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   59.80%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   60.30%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   60.80%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   61.31%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   61.81%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   62.31%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   62.81%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   63.32%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   63.82%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   64.32%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   64.82%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   65.33%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   65.83%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   66.33%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2   66.83%  Refine tree\r",
      "00:00:00    38 MB(-2%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:00    38 MB(-2%)  Iter   2    0.50%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    1.00%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    1.49%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    1.99%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    2.49%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    2.99%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    3.48%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    3.98%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    4.48%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    4.98%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    5.47%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    5.97%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    6.47%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    6.97%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    7.46%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    7.96%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    8.46%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    8.96%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    9.45%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2    9.95%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   10.45%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   10.95%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   11.44%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   11.94%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   12.44%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   12.94%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   13.43%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   13.93%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   14.43%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   14.93%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   15.42%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   15.92%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   16.42%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   16.92%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   17.41%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   17.91%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   18.41%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   18.91%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   19.40%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   19.90%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   20.40%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   20.90%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   21.39%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   21.89%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   22.39%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   22.89%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   23.38%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   23.88%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   24.38%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   24.88%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   25.37%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   25.87%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   26.37%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   26.87%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   27.36%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   27.86%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   28.36%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   28.86%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   29.35%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   29.85%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   30.35%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   30.85%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   31.34%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   31.84%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   32.34%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   32.84%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   33.33%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   33.83%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   34.33%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   34.83%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   35.32%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   35.82%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   36.32%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   36.82%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   37.31%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   37.81%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   38.31%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   38.81%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   39.30%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   39.80%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   40.30%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   40.80%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   41.29%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   41.79%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   42.29%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   42.79%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   43.28%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   43.78%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   44.28%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   44.78%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   45.27%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   45.77%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   46.27%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   46.77%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   47.26%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   47.76%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   48.26%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   48.76%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   49.25%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   49.75%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   50.25%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   50.75%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   51.24%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   51.74%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   52.24%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   52.74%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   53.23%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   53.73%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   54.23%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   54.73%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   55.22%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   55.72%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   56.22%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   56.72%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   57.21%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   57.71%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   58.21%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   58.71%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   59.20%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   59.70%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   60.20%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   60.70%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   61.19%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   61.69%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   62.19%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   62.69%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   63.18%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   63.68%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   64.18%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   64.68%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   65.17%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   65.67%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   66.17%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   66.67%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   67.16%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   67.66%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   68.16%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   68.66%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   69.15%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   69.65%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   70.15%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   70.65%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   71.14%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   71.64%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   72.14%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   72.64%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   73.13%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   73.63%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   74.13%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   74.63%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   75.12%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   75.62%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   76.12%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   76.62%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   77.11%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   77.61%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   78.11%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   78.61%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   79.10%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   79.60%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   80.10%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   80.60%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   81.09%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   81.59%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   82.09%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   82.59%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   83.08%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   83.58%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   84.08%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   84.58%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   85.07%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   85.57%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   86.07%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   86.57%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   87.06%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   87.56%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   88.06%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   88.56%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   89.05%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   89.55%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   90.05%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   90.55%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   91.04%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   91.54%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   92.04%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   92.54%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   93.03%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   93.53%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   94.03%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   94.53%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   95.02%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   95.52%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   96.02%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   96.52%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   97.01%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   97.51%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   98.01%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   98.51%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   99.00%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2   99.50%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2  100.00%  Root alignment\r",
      "00:00:00    38 MB(-2%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00    38 MB(-2%)  Iter   2  100.00%  Root alignment\r",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:02    38 MB(-2%)  Iter   3  100.00%  Refine biparts\n",
      "00:00:03    38 MB(-2%)  Iter   4  100.00%  Refine biparts\n",
      "00:00:03    38 MB(-2%)  Iter   5  100.00%  Refine biparts\n",
      "00:00:03    38 MB(-2%)  Iter   5  100.00%  Refine biparts\n"
     ]
    }
   ],
   "source": [
    "!muscle -in conc_protein_seq.fa -out alig_muscle.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2ad94e9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Csa16g043530.1_Camelina_sativa Csa16g043530.1 pep chromosome:Cs:16:22269739:22270303:-1 gene:Csa16g043530 transcript:Csa16g043530.1 gene_biotype:protein_coding transcript_biotype:protein_coding\r\n",
      "---------------------------------MASNTSFLFV--TVTLLLIL-----NV\r\n",
      "SCR-----------------------TLPAA---LA---GS-------------------\r\n",
      "---TSIAARLTGGG----------LMQ-------------CWDAL-YELNLALTRSFSSS\r\n",
      "-ST-----GRPNL-----------------------------------------------\r\n",
      "-----------APVAVTPLTSLPTIAGRRCL----LLLA-LHLRKPMSSVVSVNLRL---\r\n",
      "---------PAVLLRLLPLSTFD-------TFSSLIDKDPPNWK----------------\r\n",
      "------\r\n",
      ">EOY18917_Theobroma_cacao EOY18917 pep chromosome:Theobroma_cacao_20110822:10:6160510:6171885:-1 gene:TCM_043423 transcript:EOY18917 gene_biotype:protein_coding transcript_biotype:protein_coding description:Uncharacterized protein\r\n",
      "-----------------------------------MARKNVFL--LLSMFCLL-----AA\r\n"
     ]
    }
   ],
   "source": [
    "!head alig_muscle.fa"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "ab944ec9",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_mafft.png](alig_muscle.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29630a3",
   "metadata": {},
   "source": [
    "## MAFFT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "a1d98780",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "nthread = 0\n",
      "stacksize: 8192 kb\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "  201 / 201\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "  190 / 201\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP   198 / 200 d\n",
      "Reallocating..done. *alloclen = 1587\n",
      "STEP   200 / 200 d\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "  100 / 201\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "  190 / 201\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP   197 / 200 d\n",
      "Reallocating..done. *alloclen = 1576\n",
      "STEP   200 / 200 d\n",
      "done.\n",
      "\n",
      "disttbfast (aa) Version 7.310 alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "dndpre (aa) Version 7.310 alg=X, model=BLOSUM62, 1.53, +0.12, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "\n",
      "  190 / 201\n",
      "Segment   1/  2    1- 369\n",
      "Segment   2/  2  363- 498.    identical.    rejected. identical.    identical.    rejected. rejected. rejected. rejected. rejected. identical.    identical.    rejected. identical.    identical.    rejected. rejected. rejected. identical.    rejected. rejected. identical.    accepted. rejected. rejected. rejected. identical.    accepted. accepted. accepted. rejected. rejected. rejected. identical.    rejected. identical.   \n",
      "done 002-001-1  identical.    rejected. identical.    identical.    rejected. rejected. identical.    identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.   \n",
      "dvtditr (aa) Version 7.310 alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!mafft --auto conc_protein_seq.fa > mafft/alig_mafft.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "d5cc81dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">QL10p018382_mrna_CDS_2_Quercus_lobata QL10p018382:mrna:CDS:2 pep chromosome:ValleyOak3.0:10:18380679:18384186:1 gene:QL10p018382 transcript:QL10p018382:mrna gene_biotype:protein_coding transcript_biotype:protein_coding\r\n",
      "---------------------------------MAS----------------------QN\r\n",
      "LF-VL-L---SLA-CL-LAMTN--------------------------VTAT-RD-----\r\n",
      "---LYIK------PGH-----------------------RLAT------------R----\r\n",
      "------PET-------------S--------S-E-----G-------LVP-CWNAPW-EL\r\n",
      "KSCPNEIVVFF-LN-------------GQTDI----------------------------\r\n",
      "------------------------------APDCCRAITII-THHCWPA--MLTS-LGFT\r\n",
      "AEEGNILRGYCA----A--E--------------ASV-PST-A-----PPL---------\r\n",
      "GLPL-------------------------P------------------------RA----\r\n",
      "------------------------\r\n"
     ]
    }
   ],
   "source": [
    "!head mafft/alig_mafft.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d834ffb8",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_mafft.png](alig_mafft.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "200c52c8",
   "metadata": {},
   "source": [
    "# ClustalO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b434fba",
   "metadata": {},
   "source": [
    "**1. ClustalO Alignment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "cbdc41f3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using 8 threads\n",
      "Read 201 sequences (type: Protein) from conc_protein_seq.fa\n",
      "Using 58 seeds (chosen with constant stride from length sorted seqs) for mBed (from a total of 201 sequences)\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 0.78u 0.00s 00:00:00.78 Elapsed: 00:00:00\n",
      "mBed created 4 cluster/s (with a minimum of 1 and a soft maximum of 100 sequences each)\n",
      "Distance calculation within sub-clusters done. CPU time: 1.04u 0.00s 00:00:01.04 Elapsed: 00:00:00\n",
      "Guide-tree computation (mBed) done.\n",
      "Progressive alignment progress done. CPU time: 140.25u 0.22s 00:02:20.47 Elapsed: 00:00:21\n",
      "Alignment written to clustalo/alig_clustalo.fa\n"
     ]
    }
   ],
   "source": [
    "!clustalo --infile=conc_protein_seq.fa --outfile=clustalo/alig_clustalo.fa --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "c56421b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">QL10p018382_mrna_CDS_2_Quercus_lobata QL10p018382:mrna:CDS:2 pep chromosome:ValleyOak3.0:10:18380679:18384186:1 gene:QL10p018382 transcript:QL10p018382:mrna gene_bio\r\n",
      "------------------------------------------------------------\r\n",
      "--------------MA-------SQNLFVLL-----SLACLLAM-----TNVTATRDLY-\r\n",
      "-IKP------------------------------G-HRLATR---PE----TSSEG----\r\n",
      "LVPCWNAPWEL----------------------------------------KSCPNEIVV\r\n",
      "FFLNG-------QTDIAPDCCRAI---TII-THHCWP--AMLTSLGFTAEEGNILRGYCA\r\n",
      "AEASVPSTAP------PLGLPLP--------------------RA---------------\r\n",
      "------\r\n",
      ">QL06p031114_mrna_CDS_2_Quercus_lobata QL06p031114:mrna:CDS:2 pep chromosome:ValleyOak3.0:6:31112829:31115680:-1 gene:QL06p031114 transcript:QL06p031114:mrna gene_bio\r\n",
      "------------------------------------------------------------\r\n"
     ]
    }
   ],
   "source": [
    "!head clustalo/alig_clustalo.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e171ccaa",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_clustalo.png](alig_clustalo.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40b080d2",
   "metadata": {},
   "source": [
    "**2. Using the ClustalO alignment to build a tree**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "880a3bd6",
   "metadata": {},
   "source": [
    "#### IQTREE\n",
    "\n",
    "* [IQ-Tree v2.2.0_beta](https://anaconda.org/bioconda/iqtree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "a7c0986e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.3 for Linux 64-bit built Dec 20 2020\r\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\r\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\r\n",
      "\r\n",
      "Host:    AnnaToiThink (AVX512, FMA3, 31 GB RAM)\r\n",
      "Command: iqtree -m MFP -s clustalo/alig_clustalo.fa --prefix clustalo_not_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes\r\n",
      "Seed:    948223 (Using SPRNG - Scalable Parallel Random Number Generator)\r\n",
      "Time:    Fri May 13 12:24:46 2022\r\n",
      "Kernel:  AVX+FMA - 1 threads (8 CPU cores detected)\r\n",
      "\r\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 8 cores!\r\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\r\n",
      "\r\n",
      "Reading alignment file clustalo/alig_clustalo.fa ... Fasta format detected\r\n",
      "Alignment most likely contains protein sequences\r\n",
      "Alignment has 201 sequences with 366 columns, 320 distinct patterns\r\n",
      "225 parsimony-informative, 59 singleton sites, 82 constant sites\r\n",
      "                                                     Gap/Ambiguity  Composition  p-value\r\n",
      "   1  QL10p018382_mrna_CDS_2_Quercus_lobata                 64.21%    passed     98.95%\r\n",
      "   2  QL06p031114_mrna_CDS_2_Quercus_lobata                 64.48%    passed     99.93%\r\n",
      "   3  QL10p018398_mrna_CDS_1_Quercus_lobata                 64.21%    passed     99.71%\r\n",
      "   4  QL11p027894_mrna_CDS_1_Quercus_lobata                 64.48%    passed     99.27%\r\n",
      "   5  QL10p018395_mrna_CDS_1_Quercus_lobata                 64.48%    passed     38.13%\r\n",
      "   6  QL05p056248_mrna_Quercus_lobata                       70.77%    passed     60.93%\r\n",
      "   7  QL10p018379_mrna_CDS_1_Quercus_lobata                 68.85%    passed     75.57%\r\n",
      "   8  QL06p031123_mrna_CDS_1_Quercus_lobata                 64.48%    passed     98.25%\r\n",
      "   9  PNT58448_Populus_trichocarpa                          62.57%    passed     92.54%\r\n",
      "  10  PNT57551_Populus_trichocarpa                          64.48%    passed     13.86%\r\n",
      "  11  PNT57550_Populus_trichocarpa                          60.11%    passed     97.40%\r\n",
      "  12  PNT46426_Populus_trichocarpa                          62.02%    passed     92.51%\r\n",
      "  13  PNT20170_Populus_trichocarpa                          61.75%    passed     95.85%\r\n",
      "  14  PNT12323_Populus_trichocarpa                          88.25%    passed     11.50%\r\n",
      "  15  CDS_HanXRQr2_Chr12g0556091.1_Helianthus_annuus        57.65%    passed     66.25%\r\n",
      "  16  CDS_HanXRQr2_Chr15g0699451.1_Helianthus_annuus        60.38%    passed     38.35%\r\n",
      "  17  CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus        70.77%    passed     86.83%\r\n",
      "  18  CDS_HanXRQr2_Chr02g0065251.1_Helianthus_annuus        64.21%    passed     94.69%\r\n",
      "  19  CDS_HanXRQr2_Chr15g0708181.1_Helianthus_annuus        59.56%    failed      0.01%\r\n",
      "  20  CDS_HanXRQr2_Chr04g0158231.1_Helianthus_annuus        65.03%    passed     87.31%\r\n",
      "  21  CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus        64.21%    passed     94.94%\r\n",
      "  22  Bo8g101500.1_Brassica_oleracea                        65.30%    passed     92.51%\r\n",
      "  23  Bo7g119230.1_Brassica_oleracea                        65.03%    passed     92.11%\r\n",
      "  24  Bo2g088760.1_Brassica_oleracea                        50.00%    failed      0.16%\r\n",
      "  25  Bo1g002490.1_Brassica_oleracea                        64.21%    passed     93.45%\r\n",
      "  26  Bo2g091120.1_Brassica_oleracea                        57.65%    passed     63.25%\r\n",
      "  27  Bo4g147120.1_Brassica_oleracea                        66.12%    passed     93.43%\r\n",
      "  28  Bo6g083750.1_Brassica_oleracea                        62.57%    passed     79.27%\r\n",
      "  29  MD10G1029400_Mdomestica_491_v1                        58.20%    passed     98.68%\r\n",
      "  30  MD15G1340900_Mdomestica_491_v1                        59.29%    passed     37.87%\r\n",
      "  31  MD01G1001000_Mdomestica_491_v1                        59.02%    passed     32.62%\r\n",
      "  32  RHC11H1G0771.2.1.cds.1_Solanum_tuberosum_rh8903916    61.75%    failed      3.87%\r\n",
      "  33  RHC11H1G0775.2.1.cds.1_Solanum_tuberosum_rh8903916    64.21%    passed     69.59%\r\n",
      "  34  RHC04H1G2339.2.1.cds.1_Solanum_tuberosum_rh8903916    64.48%    passed     95.76%\r\n",
      "  35  KJB60013_Gossypium_raimondii                          65.03%    passed     90.70%\r\n",
      "  36  KJB57150_Gossypium_raimondii                          62.57%    passed     48.98%\r\n",
      "  37  KJB71292_Gossypium_raimondii                          66.12%    passed     99.91%\r\n",
      "  38  KZN08747_Daucus_carota                                70.77%    passed     93.46%\r\n",
      "  39  KZN08752_Daucus_carota                                61.20%    passed     81.36%\r\n",
      "  40  ERN08734_Amborella_trichopoda                         65.57%    passed     66.88%\r\n",
      "  41  HORVU7Hr1G102010.1_Hordeum_vulgare                    64.48%    passed     18.33%\r\n",
      "  42  HORVU4Hr1G055410.2_Hordeum_vulgare                    57.10%    passed     67.73%\r\n",
      "  43  HORVU4Hr1G055410.3_Hordeum_vulgare                    65.85%    passed     92.07%\r\n",
      "  44  HORVU7Hr1G101980.1_Hordeum_vulgare                    52.46%    failed      2.53%\r\n",
      "  45  HORVU4Hr1G020010.2_Hordeum_vulgare                    69.67%    passed     10.93%\r\n",
      "  46  HORVU4Hr1G020010.1_Hordeum_vulgare                    62.02%    failed      3.40%\r\n",
      "  47  Aco010349.1.mrna1_Ananas_comosus                      21.86%    failed      0.00%\r\n",
      "  48  Aco016990.1.mrna1.cds1_Ananas_comosus                 65.85%    passed     79.76%\r\n",
      "  49  Aco029478.1.mrna1.cds1_Ananas_comosus                 56.83%    failed      3.79%\r\n",
      "  50  Aco003193.1.mrna1.cds1_Ananas_comosus                 56.83%    failed      3.79%\r\n",
      "  51  Aco019928.1.mrna1.cds1_Ananas_comosus                 77.87%    passed     27.39%\r\n",
      "  52  Aco024757.1.mrna1.cds1_Ananas_comosus                 73.77%    passed     60.91%\r\n",
      "  53  Aco024977.1.mrna1.cds1_Ananas_comosus                 72.68%    failed      0.01%\r\n",
      "  54  Aco018685.1.mrna1.cds1_Ananas_comosus                 61.48%    passed     38.29%\r\n",
      "  55  Aco031263.1.mrna1.cds1_Ananas_comosus                 86.34%    passed     72.81%\r\n",
      "  56  PEQU_28350.1_Phalaenopsis_equestris                   63.93%    passed     28.15%\r\n",
      "  57  ONH90092_Prunus_persica                               66.67%    passed     57.57%\r\n",
      "  58  ONH90042_Prunus_persica                               58.47%    passed     89.54%\r\n",
      "  59  ONH90434_Prunus_persica                               66.12%    passed     56.15%\r\n",
      "  60  ONH90087_Prunus_persica                               65.30%    passed     98.66%\r\n",
      "  61  ONH90046_Prunus_persica                               58.47%    passed     89.38%\r\n",
      "  62  ONH89505_Prunus_persica                               61.20%    passed     84.07%\r\n",
      "  63  ONI01825_Prunus_persica                               53.01%    failed      2.56%\r\n",
      "  64  Os11t0168000-01_Oryza_sativa                          60.66%    failed      2.55%\r\n",
      "  65  Os03t0296600-01_Oryza_sativa                          58.74%    passed     15.63%\r\n",
      "  66  EES16714_Sorghum_bicolor                              51.64%    failed      0.46%\r\n",
      "  67  EER95019_Sorghum_bicolor                              53.28%    failed      4.20%\r\n",
      "  68  EOX92363_Theobroma_cacao                              63.66%    passed     93.43%\r\n",
      "  69  EOY04380_Theobroma_cacao                              62.84%    passed     37.83%\r\n",
      "  70  EOY04381_Theobroma_cacao                              63.39%    passed     83.86%\r\n",
      "  71  EOY18917_Theobroma_cacao                              63.39%    passed     26.65%\r\n",
      "  72  EOY16769_Theobroma_cacao                              61.75%    passed     67.96%\r\n",
      "  73  EOY27508_Theobroma_cacao                              63.66%    passed     99.32%\r\n",
      "  74  Ma08_p15310.1_Musa_acuminata                          61.75%    passed     59.94%\r\n",
      "  75  Ma06_p27500.1_Musa_acuminata                          56.28%    passed     77.40%\r\n",
      "  76  Solyc04g071640.1.1.1_Solanum_lycopersicum             64.75%    passed     85.51%\r\n",
      "  77  Solyc11g005500.1.1.1_Solanum_lycopersicum             64.48%    passed     56.03%\r\n",
      "  78  Solyc11g005540.1.1.1_Solanum_lycopersicum             62.84%    passed      5.73%\r\n",
      "  79  itb06g24100.t1_Ipomoea_triloba                        65.57%    passed     53.76%\r\n",
      "  80  itb04g21400.t1_Ipomoea_triloba                        59.56%    passed     70.69%\r\n",
      "  81  itb05g22120.t1_Ipomoea_triloba                        52.46%    passed     49.83%\r\n",
      "  82  OIW19998_Lupinus_angustifolius                        63.39%    passed     96.05%\r\n",
      "  83  OIW19997_Lupinus_angustifolius                        63.66%    failed      0.81%\r\n",
      "  84  OIW20030_Lupinus_angustifolius                        68.85%    passed     98.07%\r\n",
      "  85  Csa16g040920.1.cds1_Camelina_sativa                   58.74%    passed     59.14%\r\n",
      "  86  Csa07g048320.1.cds1_Camelina_sativa                   58.74%    passed     72.05%\r\n",
      "  87  Csa12g002210.1.cds1_Camelina_sativa                   66.12%    passed     57.62%\r\n",
      "  88  Csa07g048380.1.cds1_Camelina_sativa                   85.52%    failed      1.88%\r\n",
      "  89  Csa10g002090.1.cds1_Camelina_sativa                   66.12%    passed     54.00%\r\n",
      "  90  Csa16g043520.1.cds1_Camelina_sativa                   65.85%    passed     72.06%\r\n",
      "  91  Csa07g051890.1.cds1_Camelina_sativa                   65.30%    passed     68.01%\r\n",
      "  92  Csa09g081500.1.cds1_Camelina_sativa                   58.47%    passed     71.64%\r\n",
      "  93  Csa09g086140.1.cds1_Camelina_sativa                   65.85%    passed     82.46%\r\n",
      "  94  Csa11g002350.1.cds1_Camelina_sativa                   66.12%    passed     46.93%\r\n",
      "  95  Csa09g086150.1.cds1_Camelina_sativa                   65.57%    passed     78.57%\r\n",
      "  96  Csa27089s010.1.cds1_Camelina_sativa                   84.15%    passed     28.73%\r\n",
      "  97  Csa16g043530.1_Camelina_sativa                        61.48%    failed      2.04%\r\n",
      "  98  Csa07g051880.1_Camelina_sativa                        73.77%    passed     72.33%\r\n",
      "  99  FCD_00011687-RA_cds_Ficus_carica                      67.76%    passed     95.57%\r\n",
      " 100  FCD_00028685-RA_cds_Ficus_carica                      62.57%    passed     93.67%\r\n",
      " 101  FCD_00030006-RA_cds_Ficus_carica                      62.57%    passed     92.48%\r\n",
      " 102  FCD_00011685-RA_cds_Ficus_carica                      58.47%    passed     61.08%\r\n",
      " 103  RZC48302_Papaver_somniferum                           56.56%    passed     88.62%\r\n",
      " 104  RZC49191_Papaver_somniferum                           60.11%    passed     84.67%\r\n",
      " 105  RZC49225_Papaver_somniferum                           57.38%    passed     66.92%\r\n",
      " 106  RZC55249_Papaver_somniferum                           57.38%    passed     67.34%\r\n",
      " 107  RZC55281_Papaver_somniferum                           58.47%    passed     70.07%\r\n",
      " 108  RZC57082_Papaver_somniferum                           61.48%    passed     44.08%\r\n",
      " 109  RZC57075_Papaver_somniferum                           77.60%    passed     71.76%\r\n",
      " 110  RZC57081_Papaver_somniferum                           56.83%    passed     98.01%\r\n",
      " 111  RZC62100_Papaver_somniferum                           58.74%    passed     99.27%\r\n",
      " 112  RZC62099_Papaver_somniferum                           58.47%    passed     99.87%\r\n",
      " 113  RZC68710_Papaver_somniferum                           63.66%    passed     99.23%\r\n",
      " 114  RZC68605_Papaver_somniferum                           70.22%    failed      0.02%\r\n",
      " 115  RZC68602_Papaver_somniferum                           58.74%    passed     99.87%\r\n",
      " 116  PRQ49244_Rosa_chinensis                               72.13%    passed     82.51%\r\n",
      " 117  PRQ49241_Rosa_chinensis                               63.11%    passed     59.51%\r\n",
      " 118  PRQ25531_Rosa_chinensis                               59.84%    passed     69.30%\r\n",
      " 119  PRQ24668_Rosa_chinensis                               49.18%    passed      8.24%\r\n",
      " 120  PRQ25532_Rosa_chinensis                               60.11%    passed     70.17%\r\n",
      " 121  PRQ17678_Rosa_chinensis                               63.39%    passed     99.36%\r\n",
      " 122  Vradi07g02140.1_Vigna_radiata                         63.66%    passed     51.43%\r\n",
      " 123  Vradi01g10860.1_Vigna_radiata                         68.31%    passed     34.57%\r\n",
      " 124  OIT31467_Nicotiana_attenuata                          65.57%    passed     51.85%\r\n",
      " 125  ESW31795_Phaseolus_vulgaris                           70.49%    passed     51.99%\r\n",
      " 126  ESW18393_Phaseolus_vulgaris                           64.75%    passed     70.64%\r\n",
      " 127  ESW28268_Phaseolus_vulgaris                           62.30%    passed     97.70%\r\n",
      " 128  KEH21230_Medicago_truncatula                          61.48%    passed     22.52%\r\n",
      " 129  AES70181_Medicago_truncatula                          60.38%    passed     71.34%\r\n",
      " 130  AES70191_Medicago_truncatula                          65.03%    passed     23.80%\r\n",
      " 131  OAY36032_Manihot_esculenta                            63.11%    passed     98.36%\r\n",
      " 132  OAY36033_Manihot_esculenta                            58.74%    passed     70.90%\r\n",
      " 133  KRH63270_Glycine_max                                  62.30%    passed     70.80%\r\n",
      " 134  KRH59512_Glycine_max                                  67.21%    passed     43.84%\r\n",
      " 135  KRH54635_Glycine_max                                  62.57%    passed     82.46%\r\n",
      " 136  KRH43377_Glycine_max                                  67.21%    passed     75.61%\r\n",
      " 137  KRH39377_Glycine_max                                  63.66%    passed     49.01%\r\n",
      " 138  KRH35123_Glycine_max                                  81.15%    passed     68.15%\r\n",
      " 139  KRG98863_Glycine_max                                  69.95%    passed     50.10%\r\n",
      " 140  KRG91676_Glycine_max                                  61.48%    passed     77.03%\r\n",
      " 141  KMT04640_Beta_vulgaris                                61.20%    passed     81.74%\r\n",
      " 142  KMT16083_Beta_vulgaris                                68.58%    passed     34.62%\r\n",
      " 143  DRNTG_09031.1.cds1_Dioscorea_rotundata                76.23%    passed     62.88%\r\n",
      " 144  FtPinG0007730800.01.T01_Fagopyrum_tataricum           72.40%    passed     58.97%\r\n",
      " 145  KGN64838_Cucumis_sativus                              61.20%    passed     74.83%\r\n",
      " 146  KGN64837_Cucumis_sativus                              68.31%    passed     54.82%\r\n",
      " 147  KGN46138_Cucumis_sativus                              63.66%    passed     79.67%\r\n",
      " 148  cds.evm.model.01.1369_Cannabis_sativa_female          62.84%    passed     28.48%\r\n",
      " 149  cds.evm.model.04.1284_Cannabis_sativa_female          56.01%    passed     78.89%\r\n",
      " 150  cds.Jr15_01190_p1_Juglans_regia                       57.38%    passed     70.41%\r\n",
      " 151  VIT_09s0002g01570.t01_Vitis_vinifera                  65.03%    passed     64.58%\r\n",
      " 152  VIT_07s0031g02070.t01_Vitis_vinifera                  63.11%    passed     97.25%\r\n",
      " 153  OE9A117531C1_Olea_europaea                            63.11%    passed     64.33%\r\n",
      " 154  NC1G0017660.1_cds_Nymphaea_colorata                   65.85%    passed     58.40%\r\n",
      " 155  NC11G0118510.1_cds_Nymphaea_colorata                  65.03%    passed     14.65%\r\n",
      " 156  NC4G0237690.1_cds_Nymphaea_colorata                   83.88%    passed     34.48%\r\n",
      " 157  Bradi2g28070.1.p_Bdistachyon_314_v3                   60.66%    passed     30.48%\r\n",
      " 158  Bradi4g24640.1.p_Bdistachyon_314_v3                   56.28%    passed      5.01%\r\n",
      " 159  Bradi1g65190.1.p_Bdistachyon_314_v3                   52.73%    failed      1.12%\r\n",
      " 160  ESR47720_Citrus_clementina                            60.93%    passed     99.99%\r\n",
      " 161  Zm00001eb396810_P001_Zea_mays                         56.28%    passed     14.97%\r\n",
      " 162  FvH4_2g18700.t1_Fvesca_677_v4                         59.56%    passed     30.69%\r\n",
      " 163  FvH4_2g18910.t1_Fvesca_677_v4                         60.66%    passed     91.52%\r\n",
      " 164  FvH4_2g13400.t1_Fvesca_677_v4                         57.65%    passed     87.84%\r\n",
      " 165  FvH4_2g13390.t1_Fvesca_677_v4                         57.92%    passed     86.28%\r\n",
      " 166  FvH4_2g18870.t1_Fvesca_677_v4                         62.30%    passed     52.76%\r\n",
      " 167  FvH4_4g25660.t1_Fvesca_677_v4                         63.66%    passed     62.34%\r\n",
      " 168  FvH4_1g23830.t1_Fvesca_677_v4                         61.20%    passed     39.13%\r\n",
      " 169  FvH4_5g02500.t1_Fvesca_677_v4                         59.84%    passed     85.86%\r\n",
      " 170  Kaladp0040s0523.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   63.93%    passed     89.14%\r\n",
      " 171  Kaladp0026s0134.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   62.02%    passed     95.65%\r\n",
      " 172  Kaladp0515s0048.1.v1.1_Kalanchoe_fedtschenkoi         69.40%    failed      0.00%\r\n",
      " 173  Kaladp0033s0187.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   63.39%    passed     84.63%\r\n",
      " 174  ESQ53189_Eutrema_salsugineum                          65.03%    passed     64.16%\r\n",
      " 175  ESQ46086_Eutrema_salsugineum                          72.95%    passed     51.15%\r\n",
      " 176  ESQ46085_Eutrema_salsugineum                          65.85%    passed     84.69%\r\n",
      " 177  ESQ27444_Eutrema_salsugineum                          56.56%    passed     40.64%\r\n",
      " 178  SIN_1014976.t.cds1_Sesamum_indicum                    64.48%    passed     62.61%\r\n",
      " 179  SIN_1019387.t.cds1_Sesamum_indicum                    63.11%    passed     80.68%\r\n",
      " 180  CDS_MD10G0023400.1_Malus_domestica_golden             58.20%    passed     98.68%\r\n",
      " 181  CDS_MD01G0004200.1_Malus_domestica_golden             59.02%    passed     32.62%\r\n",
      " 182  TraesCS4D02G109400.1.cds1_Triticum_aestivum           57.92%    failed      0.79%\r\n",
      " 183  TraesCS7D02G440400.1.cds1_Triticum_aestivum           54.64%    passed     11.22%\r\n",
      " 184  TraesCS7A02G451500.1.cds1_Triticum_aestivum           55.19%    passed     15.49%\r\n",
      " 185  TraesCS7A02G451600.1.cds1_Triticum_aestivum           58.74%    passed      8.05%\r\n",
      " 186  TraesCS7B02G351400.1.cds1_Triticum_aestivum           59.02%    passed     11.07%\r\n",
      " 187  TraesCS7B02G351300.1.cds1_Triticum_aestivum           54.10%    passed     33.21%\r\n",
      " 188  TraesCS4B02G190500.1.cds1_Triticum_aestivum           54.92%    passed     22.84%\r\n",
      " 189  TraesCS7D02G440500.1.cds1_Triticum_aestivum           58.74%    passed      5.33%\r\n",
      " 190  TraesCS4B02G111900.1_Triticum_aestivum                60.38%    failed      0.01%\r\n",
      " 191  TraesCS4A02G208200.1.cds1_Triticum_aestivum           57.92%    failed      0.67%\r\n",
      " 192  TraesCS4A02G113600.1.cds1_Triticum_aestivum           57.10%    passed     24.15%\r\n",
      " 193  AT4G39340.1_Arabidopsis_thaliana                      65.30%    passed     71.41%\r\n",
      " 194  AT2G21750.1_Arabidopsis_thaliana                      65.85%    passed     77.25%\r\n",
      " 195  AT1G76750.1_Arabidopsis_thaliana                      56.83%    passed     63.87%\r\n",
      " 196  AT2G21740.1_Arabidopsis_thaliana                      65.85%    passed     88.50%\r\n",
      " 197  cds-PLY89992.1_Lactuca_sativa                         59.84%    passed     14.07%\r\n",
      " 198  cds-PLY89774.1_Lactuca_sativa                         64.21%    passed     33.04%\r\n",
      " 199  CDP12837_Coffea_canephora                             57.92%    passed     47.99%\r\n",
      " 200  PSS26677_Actinidia_chinensis                          62.57%    passed     98.78%\r\n",
      " 201  jg50396.t1_Fagopyrum_esculentum                       67.49%    passed     74.87%\r\n",
      "WARNING: 198 sequences contain more than 50% gaps/ambiguity\r\n",
      "****  TOTAL                                                 62.98%  22 sequences failed composition chi2 test (p-value<5%; df=19)\r\n",
      "NOTE: CDS_MD10G0023400.1_Malus_domestica_golden is identical to MD10G1029400_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD01G0004200.1_Malus_domestica_golden is identical to MD01G1001000_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: Aco003193.1.mrna1.cds1_Ananas_comosus is identical to Aco029478.1.mrna1.cds1_Ananas_comosus but kept for subsequent analysis\r\n",
      "\r\n",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.139 seconds\n",
      "Perform fast likelihood tree search using LG+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -30786.685\n",
      "Optimal log-likelihood: -30785.894\n",
      "Proportion of invariable sites: 0.001\n",
      "Gamma shape alpha: 1.265\n",
      "Parameters optimization took 1 rounds (2.162 sec)\n",
      "Time for fast ML tree search: 14.436 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 106 MB RAM!\n",
      "ModelFinder will test up to 546 protein models (sample size: 366) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            32184.915    399 65167.829    384367.829   66724.980\n",
      "  2  LG+I          32184.898    400 65169.796    385969.796   66730.849\n",
      "  3  LG+G4         30785.709    400 62371.417    383171.417   63932.471\n",
      "  4  LG+I+G4       30785.723    401 62373.446    384777.446   63938.402\n",
      "  5  LG+R2         30950.591    401 62703.183    385107.183   64268.139\n",
      "  6  LG+R3         30800.223    403 62406.446    388030.446   63979.207\n",
      "  7  LG+R4         30741.471    405 62292.943    391152.943   63873.509\n",
      "  8  LG+R5         30721.892    407 62257.783    394369.783   63846.155\n",
      "  9  LG+R6         30711.126    409 62240.252    397620.252   63836.429\n",
      " 10  LG+R7         30710.942    411 62243.885    400907.885   63847.867\n",
      " 21  LG+F+R5       30758.684    426 62369.368    426173.368   64031.890\n",
      " 22  LG+F+R6       30748.592    428 62353.183    429577.183   64023.510\n",
      " 34  WAG+R5        30523.033    407 61860.067    393972.067   63448.439\n",
      " 35  WAG+R6        30519.084    409 61856.167    397236.167   63452.345\n",
      " 47  WAG+F+R5      30481.769    426 61815.537    425619.537   63478.059\n",
      " 48  WAG+F+R6      30477.749    428 61811.497    429035.497   63481.825\n",
      " 60  JTT+R5        30262.255    407 61338.511    393450.511   62926.882\n",
      " 61  JTT+R6        30259.096    409 61336.192    396716.192   62932.369\n",
      " 73  JTT+F+R5      30366.910    426 61585.819    425389.819   63248.341\n",
      " 74  JTT+F+R6      30362.954    428 61581.908    428805.908   63252.235\n",
      " 86  JTTDCMut+R5   30271.616    407 61357.231    393469.231   62945.603\n",
      " 87  JTTDCMut+R6   30268.273    409 61354.546    396734.546   62950.723\n",
      " 99  JTTDCMut+F+R5 30369.686    426 61591.373    425395.373   63253.894\n",
      "100  JTTDCMut+F+R6 30365.391    428 61586.782    428810.782   63257.109\n",
      "112  DCMut+R5      30710.755    407 62235.510    394347.510   63823.882\n",
      "113  DCMut+R6      30707.943    409 62233.886    397613.886   63830.063\n",
      "125  DCMut+F+R5    30621.951    426 62095.903    425899.903   63758.424\n",
      "126  DCMut+F+R6    30618.137    428 62092.273    429316.273   63762.600\n",
      "138  VT+R5         30530.178    407 61874.356    393986.356   63462.728\n",
      "139  VT+R6         30528.516    409 61875.031    397255.031   63471.208\n",
      "151  VT+F+R5       30567.549    426 61987.098    425791.098   63649.620\n",
      "152  VT+F+R6       30565.186    428 61986.371    429210.371   63656.698\n",
      "164  PMB+R5        30845.481    407 62504.962    394616.962   64093.334\n",
      "165  PMB+R6        30843.831    409 62505.661    397885.661   64101.838\n",
      "177  PMB+F+R5      30795.515    426 62443.031    426247.031   64105.552\n",
      "178  PMB+F+R6      30793.326    428 62442.653    429666.653   64112.980\n",
      "190  Blosum62+R5   30996.928    407 62807.856    394919.856   64396.228\n",
      "191  Blosum62+R6   30993.710    409 62805.420    398185.420   64401.597\n",
      "203  Blosum62+F+R5 30865.464    426 62582.928    426386.928   64245.450\n",
      "204  Blosum62+F+R6 30862.304    428 62580.608    429804.608   64250.935\n",
      "216  Dayhoff+R5    30710.597    407 62235.194    394347.194   63823.566\n",
      "217  Dayhoff+R6    30707.748    409 62233.496    397613.496   63829.673\n",
      "229  Dayhoff+F+R5  30622.584    426 62097.168    425901.168   63759.690\n",
      "230  Dayhoff+F+R6  30618.711    428 62093.421    429317.421   63763.748\n",
      "242  mtREV+R5      31421.062    407 63656.123    395768.123   65244.495\n",
      "243  mtREV+R6      31411.861    409 63641.721    399021.721   65237.899\n",
      "255  mtREV+F+R5    30811.257    426 62474.513    426278.513   64137.035\n",
      "256  mtREV+F+R6    30799.309    428 62454.617    429678.617   64124.944\n",
      "268  mtART+R5      32487.854    407 65789.708    397901.708   67378.079\n",
      "269  mtART+R6      32428.575    409 65675.151    401055.151   67271.328\n",
      "281  mtART+F+R5    31808.401    426 64468.802    428272.802   66131.324\n",
      "282  mtART+F+R6    31775.854    428 64407.709    431631.709   66078.036\n",
      "294  mtZOA+R5      31598.453    407 64010.906    396122.906   65599.278\n",
      "295  mtZOA+R6      31552.520    409 63923.041    399303.041   65519.218\n",
      "307  mtZOA+F+R5    31310.962    426 63473.924    427277.924   65136.445\n",
      "308  mtZOA+F+R6    31274.819    428 63405.638    430629.638   65075.966\n",
      "320  mtMet+R5      31532.503    407 63879.005    395991.005   65467.377\n",
      "321  mtMet+R6      31523.651    409 63865.302    399245.302   65461.479\n",
      "333  mtMet+F+R5    30750.674    426 62353.348    426157.348   64015.870\n",
      "334  mtMet+F+R6    30736.901    428 62329.801    429553.801   64000.128\n",
      "346  mtVer+R5      31364.842    407 63543.684    395655.684   65132.056\n",
      "347  mtVer+R6      31352.278    409 63522.556    398902.556   65118.733\n",
      "359  mtVer+F+R5    30804.569    426 62461.138    426265.138   64123.660\n",
      "360  mtVer+F+R6    30789.158    428 62434.317    429658.317   64104.644\n",
      "372  mtInv+R5      31988.822    407 64791.644    396903.644   66380.016\n",
      "373  mtInv+R6      31981.464    409 64780.929    400160.929   66377.106\n",
      "385  mtInv+F+R5    30755.289    426 62362.578    426166.578   64025.100\n",
      "386  mtInv+F+R6    30743.264    428 62342.527    429566.527   64012.854\n",
      "398  mtMAM+R5      31833.021    407 64480.042    396592.042   66068.414\n",
      "399  mtMAM+R6      31813.799    409 64445.598    399825.598   66041.776\n",
      "411  mtMAM+F+R5    31101.568    426 63055.136    426859.136   64717.658\n",
      "412  mtMAM+F+R6    31089.489    428 63034.979    430258.979   64705.306\n",
      "424  HIVb+R5       30918.824    407 62651.648    394763.648   64240.020\n",
      "425  HIVb+R6       30911.598    409 62641.195    398021.195   64237.372\n",
      "437  HIVb+F+R5     30672.203    426 62196.405    426000.405   63858.927\n",
      "438  HIVb+F+R6     30662.382    428 62180.764    429404.764   63851.091\n",
      "450  HIVw+R5       31733.448    407 64280.896    396392.896   65869.268\n",
      "451  HIVw+R6       31728.640    409 64275.281    399655.281   65871.458\n",
      "463  HIVw+F+R5     31067.519    426 62987.037    426791.037   64649.559\n",
      "464  HIVw+F+R6     31063.404    428 62982.809    430206.809   64653.136\n",
      "476  FLU+R5        30862.475    407 62538.950    394650.950   64127.322\n",
      "477  FLU+R6        30853.149    409 62524.297    397904.297   64120.474\n",
      "489  FLU+F+R5      30496.951    426 61845.902    425649.902   63508.424\n",
      "490  FLU+F+R6      30487.773    428 61831.546    429055.546   63501.873\n",
      "502  rtREV+R5      31071.358    407 62956.716    395068.716   64545.088\n",
      "503  rtREV+R6      31062.809    409 62943.618    398323.618   64539.795\n",
      "515  rtREV+F+R5    30980.893    426 62813.786    426617.786   64476.308\n",
      "516  rtREV+F+R6    30968.349    428 62792.698    430016.698   64463.025\n",
      "528  cpREV+R5      30670.180    407 62154.360    394266.360   63742.732\n",
      "529  cpREV+R6      30667.289    409 62152.579    397532.579   63748.756\n",
      "541  cpREV+F+R5    30706.894    426 62265.788    426069.788   63928.310\n",
      "542  cpREV+F+R6    30693.202    428 62242.403    429466.403   63912.730\n",
      "Akaike Information Criterion:           JTT+R6\n",
      "Corrected Akaike Information Criterion: LG+G4\n",
      "Bayesian Information Criterion:         JTT+R5\n",
      "Best-fit model: JTT+R5 chosen according to BIC\n",
      "\n",
      "All model information printed to clustalo_not_trimmed_iqtree_bootstrap.model.gz\n",
      "CPU time for ModelFinder: 3145.448 seconds (0h:52m:25s)\n",
      "Wall-clock time for ModelFinder: 3149.976 seconds (0h:52m:29s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 948223)...\n",
      "\n",
      "NOTE: 54 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -30262.255\n",
      "Optimal log-likelihood: -30262.232\n",
      "Site proportion and rates:  (0.085,0.112) (0.129,0.265) (0.154,0.543) (0.224,1.040) (0.407,1.569)\n",
      "Parameters optimization took 1 rounds (3.360 sec)\n",
      "Computing ML distances based on estimated model parameters... 5.115 sec\n",
      "WARNING: Some pairwise ML distances are too long (saturated)\n",
      "Computing BIONJ tree...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.017 seconds\n",
      "Log-likelihood of BIONJ tree: -30470.391\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 9.786 second\n",
      "Computing log-likelihood of 98 initial trees ... 31.553 seconds\n",
      "Current best score: -30262.232\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -30219.688\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 3: -30212.527\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 5: -30203.099\n",
      "Iteration 10 / LogL: -30232.969 / Time: 0h:2m:43s\n",
      "Iteration 20 / LogL: -30251.145 / Time: 0h:4m:10s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -30203.099 / CPU time: 237.760\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 30: -30197.892\n",
      "Iteration 30 / LogL: -30197.892 / Time: 0h:5m:44s (0h:19m:47s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 32: -30192.077\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 40: -30187.448\n",
      "Iteration 40 / LogL: -30187.448 / Time: 0h:7m:24s (0h:18m:58s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 44: -30187.111\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 50: -30185.399\n",
      "Iteration 50 / LogL: -30185.399 / Time: 0h:8m:59s (0h:18m:20s left)\n",
      "Log-likelihood cutoff on original alignment: -30267.717\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 56: -30183.131\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 59: -30182.668\n",
      "Iteration 60 / LogL: -30183.638 / Time: 0h:10m:42s (0h:17m:57s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 62: -30182.125\n",
      "Iteration 70 / LogL: -30182.358 / Time: 0h:12m:15s (0h:16m:21s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 76: -30182.008\n",
      "Iteration 80 / LogL: -30187.102 / Time: 0h:13m:43s (0h:16m:41s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 86: -30181.717\n",
      "Iteration 90 / LogL: -30342.262 / Time: 0h:15m:16s (0h:16m:28s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 97: -30181.649\n",
      "Iteration 100 / LogL: -30185.451 / Time: 0h:16m:44s (0h:16m:24s left)\n",
      "Log-likelihood cutoff on original alignment: -30265.563\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.986\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 102: -30181.475\n",
      "Iteration 110 / LogL: -30184.574 / Time: 0h:18m:19s (0h:15m:27s left)\n",
      "Iteration 120 / LogL: -30189.741 / Time: 0h:19m:48s (0h:13m:38s left)\n",
      "Iteration 130 / LogL: -30183.862 / Time: 0h:21m:18s (0h:11m:53s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 139: -30181.410\n",
      "Iteration 140 / LogL: -30239.137 / Time: 0h:22m:55s (0h:16m:19s left)\n",
      "Iteration 150 / LogL: -30189.178 / Time: 0h:24m:24s (0h:14m:35s left)\n",
      "Log-likelihood cutoff on original alignment: -30265.563\n",
      "Iteration 160 / LogL: -30188.269 / Time: 0h:25m:50s (0h:12m:50s left)\n",
      "Iteration 170 / LogL: -30235.371 / Time: 0h:27m:20s (0h:11m:9s left)\n",
      "Iteration 180 / LogL: -30183.510 / Time: 0h:28m:49s (0h:9m:30s left)\n",
      "Iteration 190 / LogL: -30184.149 / Time: 0h:30m:13s (0h:7m:50s left)\n",
      "Iteration 200 / LogL: -30227.868 / Time: 0h:31m:41s (0h:6m:12s left)\n",
      "Log-likelihood cutoff on original alignment: -30265.563\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.999\n",
      "Iteration 210 / LogL: -30195.401 / Time: 0h:33m:14s (0h:14m:18s left)\n",
      "Iteration 220 / LogL: -30197.338 / Time: 0h:34m:43s (0h:12m:41s left)\n",
      "Iteration 230 / LogL: -30190.220 / Time: 0h:36m:16s (0h:11m:5s left)\n",
      "Iteration 240 / LogL: -30186.227 / Time: 0h:37m:37s (0h:9m:26s left)\n",
      "TREE SEARCH COMPLETED AFTER 240 ITERATIONS / Time: 0h:37m:37s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -30181.410\n",
      "Optimal log-likelihood: -30181.409\n",
      "Site proportion and rates:  (0.085,0.110) (0.130,0.269) (0.150,0.536) (0.221,1.025) (0.415,1.565)\n",
      "Parameters optimization took 1 rounds (2.563 sec)\n",
      "BEST SCORE FOUND : -30181.409\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 1000 replicates...\n",
      "Testing tree branches by aBayes parametric test...\n",
      "3.775 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file clustalo_not_trimmed_iqtree_bootstrap.splits.nex\n",
      "Total tree length: 87.101\n",
      "\n",
      "Total number of iterations: 240\n",
      "CPU time used for tree search: 2241.187 sec (0h:37m:21s)\n",
      "Wall-clock time used for tree search: 2244.206 sec (0h:37m:24s)\n",
      "Total CPU time used: 2261.308 sec (0h:37m:41s)\n",
      "Total wall-clock time used: 2264.373 sec (0h:37m:44s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file clustalo_not_trimmed_iqtree_bootstrap.splits.nex...\n",
      "201 taxa and 1448 splits.\n",
      "Consensus tree written to clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "Reading input trees file clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "Log-likelihood of consensus tree: -30182.829\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                clustalo_not_trimmed_iqtree_bootstrap.iqtree\n",
      "  Maximum-likelihood tree:       clustalo_not_trimmed_iqtree_bootstrap.treefile\n",
      "  Likelihood distances:          clustalo_not_trimmed_iqtree_bootstrap.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          clustalo_not_trimmed_iqtree_bootstrap.splits.nex\n",
      "  Consensus tree:                clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "  Screen log file:               clustalo_not_trimmed_iqtree_bootstrap.log\n",
      "\n",
      "Date and Time: Fri May 13 13:55:04 2022\n"
     ]
    }
   ],
   "source": [
    "!iqtree -m MFP -s clustalo/alig_clustalo.fa --prefix clustalo_not_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
