{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fd650268",
   "metadata": {},
   "source": [
    "# Phylogenetic analysis\n",
    "\n",
    "For phylogenetic analysis we use the file with amino acid FASTA sequences generated in the folder `OrthoGroups_analysis`.\n",
    "\n",
    "Versions:\n",
    "* MUSCLE - v5.1\n",
    "* MAFFT - v7.505\n",
    "* ClustalO - 1.2.3\n",
    "* Ugene - v41\n",
    "* [IQ-Tree v2.0.3](https://anaconda.org/bioconda/iqtree)\n",
    "\n",
    "Coverage is universally bad, but we decided to use ClustalO to build the tree."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c57063e5",
   "metadata": {},
   "source": [
    "## MUSCLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3cf915af",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "muscle 5.1.linux64 []  32.5Gb RAM, 8 cores\n",
      "Built Feb 24 2022 03:16:15\n",
      "(C) Copyright 2004-2021 Robert C. Edgar.\n",
      "https://drive5.com\n",
      "\n",
      "Input: 201 seqs, avg length 136, max 286\n",
      "\n",
      "00:00 10Mb   CPU has 8 cores, running 8 threads\n",
      "00:17 556Mb   100.0% Calc posteriors\n",
      "00:57 570Mb   100.0% Consistency (1/2)2)\n",
      "01:50 571Mb   100.0% Consistency (2/2)/2)\n",
      "01:50 571Mb   100.0% UPGMA5           \n",
      "01:55 588Mb   100.0% Refining\n"
     ]
    }
   ],
   "source": [
    "!muscle -align conc_protein_seq.fa -output alig_muscle.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2ad94e9e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">FCD_00028685-RA_cds_Ficus_carica FCD_00028685-RA:cds pep primary_assembly:UNIPI_FiCari_1.0:10:20318531:20318944:-1 gene:FCD_00028685 transcript:FCD_00028685-RA gene_biotype:protein_coding transcript_biotype:protein_coding description:egg cell-secreted protein 1.1\r\n",
      "----------------------------------MAYT--Q---------------------M-----------------\r\n",
      "-----------------------IN-LI----------IL-LTTL-LA-L-VI--I----------------PA------\r\n",
      "--------S------------------A--T---R-PGA-ST--------------------------PSTGL-NL-VA-\r\n",
      "RL-----------K-----------------------L------------ED----------------------------\r\n",
      "-----------------E-ESSNCWD---------------------------------------SLIQ-LQACTGEVIL\r\n",
      "FF-L------------------------------------------NG---------------ET-----YLGPGCCDAI\r\n",
      "RTI----G-QHCW---------------------P--A--MFGS-LGFTTEETDVLQGYCTGE-ARGSA-----------\r\n",
      "----SP------------------P-------P----LSP----------------------------------------\r\n",
      "---------------------------------------P-------SP-P----------SI\r\n"
     ]
    }
   ],
   "source": [
    "!head alig_muscle.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab944ec9",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_muscle.png](alig_muscle.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29630a3",
   "metadata": {},
   "source": [
    "## MAFFT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a1d98780",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "  201 / 201\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "  190 / 201\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP   198 / 200  h\n",
      "Reallocating..done. *alloclen = 1610\n",
      "STEP   200 / 200  h\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "  100 / 201\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "  190 / 201\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP   199 / 200  h\n",
      "Reallocating..done. *alloclen = 1691\n",
      "STEP   200 / 200  h\n",
      "done.\n",
      "\n",
      "disttbfast (aa) Version 7.505\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "rescale = 1\n",
      "dndpre (aa) Version 7.505\n",
      "alg=X, model=BLOSUM62, 1.53, +0.12, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "rescale = 1\n",
      "\n",
      "  190 / 201\n",
      "Segment   1/  2    1- 354\n",
      "Segment   2/  2  358- 499.    identical.    identical.    rejected. rejected. rejected. rejected. rejected. rejected. accepted. rejected. rejected. rejected. accepted. rejected. rejected. identical.    rejected. identical.    identical.    rejected. accepted. identical.    rejected. rejected. identical.    rejected. rejected. rejected. rejected. rejected. rejected. rejected. rejected. rejected. identical.    rejected. identical.    identical.    rejected. identical.    identical.    identical.   \n",
      "done 002-001-1  identical.    identical.    rejected. accepted. rejected. rejected. rejected. identical.    rejected. rejected. identical.    rejected. identical.    identical.    rejected. accepted. identical.   \n",
      "dvtditr (aa) Version 7.505\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!mafft --auto conc_protein_seq.fa > alig_mafft.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d5cc81dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">QL10p018382_mrna_CDS_2_Quercus_lobata QL10p018382:mrna:CDS:2 pep chromosome:ValleyOak3.0:10:18380679:18384186:1 gene:QL10p018382 transcript:QL10p018382:mrna gene_biotype:protein_coding transcript_biotype:protein_coding\r\n",
      "---------------------------------MAS------------QNLF-VL-L-SL\r\n",
      "ACL-LAM---------TNVT---------------------AT-RD--------LYIK--\r\n",
      "------------------------P-GH-----------------RLATR----------\r\n",
      "P-ET---------------------S----S-EGLVP------------------CWNAP\r\n",
      "W-----ELKSCPNEIVVFF-LN------------------GQTD-------IAPDCCRAI\r\n",
      "TII------------------------------------THHCWPA--MLTSLGFTAEEG\r\n",
      "NILRGYC-A-------A-EASVP-----------ST----A-----PP----LG------\r\n",
      "-LPL----------------------PRA-------------------------------\r\n",
      "---------------\r\n"
     ]
    }
   ],
   "source": [
    "!head alig_mafft.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d834ffb8",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_mafft.png](alig_mafft.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "200c52c8",
   "metadata": {},
   "source": [
    "# ClustalO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b434fba",
   "metadata": {},
   "source": [
    "**1. ClustalO Alignment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cbdc41f3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using 8 threads\n",
      "Read 201 sequences (type: Protein) from conc_protein_seq.fa\n",
      "Using 58 seeds (chosen with constant stride from length sorted seqs) for mBed (from a total of 201 sequences)\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 0.45u 0.01s 00:00:00.46 Elapsed: 00:00:00\n",
      "mBed created 4 cluster/s (with a minimum of 1 and a soft maximum of 100 sequences each)\n",
      "Distance calculation within sub-clusters done. CPU time: 0.31u 0.00s 00:00:00.31 Elapsed: 00:00:00\n",
      "Guide-tree computation (mBed) done.\n",
      "Progressive alignment progress done. CPU time: 13.54u 0.06s 00:00:13.60 Elapsed: 00:00:02\n",
      "Alignment written to alig_clustalo.fa\n"
     ]
    }
   ],
   "source": [
    "!clustalo --infile=conc_protein_seq.fa --outfile=alig_clustalo.fa --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c56421b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">QL10p018382_mrna_CDS_2_Quercus_lobata QL10p018382:mrna:CDS:2 pep chromosome:ValleyOak3.0:10:18380679:18384186:1 gene:QL10p018382 transcript:QL10p018382:mrna gene_bio\r\n",
      "------------------------------------------------------------\r\n",
      "------------------------------------------MA-------SQ---NLFV\r\n",
      "LL--------SLACLLAM-----TNVTAT---RDLY--IKP------------G-HRLAT\r\n",
      "R---P-------------------ETSSEGLVPCWNAPWELKSCPNE-------------\r\n",
      "---------------------------IVVFF-LNG-------QTDIAPDCCRA---ITI\r\n",
      "I-THHCWPA--ML--------TSLGFTAEEGNILRG-----YCAAEASVPST--APPLG-\r\n",
      "------------LPLP-----------------------RA-------------------\r\n",
      "--------------\r\n",
      ">QL06p031114_mrna_CDS_2_Quercus_lobata QL06p031114:mrna:CDS:2 pep chromosome:ValleyOak3.0:6:31112829:31115680:-1 gene:QL06p031114 transcript:QL06p031114:mrna gene_bio\r\n"
     ]
    }
   ],
   "source": [
    "!head alig_clustalo.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e171ccaa",
   "metadata": {},
   "source": [
    "Alignment coverage as visualized in UGENE.\n",
    "\n",
    "![alig_clustalo.png](alig_clustalo.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40b080d2",
   "metadata": {},
   "source": [
    "**2. Using the ClustalO alignment to build a tree**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "880a3bd6",
   "metadata": {},
   "source": [
    "#### IQTREE\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a7c0986e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.3 for Linux 64-bit built Dec 20 2020\r\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\r\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\r\n",
      "\r\n",
      "Host:    AnnaToiThink (AVX512, FMA3, 31 GB RAM)\r\n",
      "Command: iqtree -m MFP -s alig_clustalo.fa --prefix clustalo_not_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes -redo\r\n",
      "Seed:    927732 (Using SPRNG - Scalable Parallel Random Number Generator)\r\n",
      "Time:    Sat May 21 15:53:50 2022\r\n",
      "Kernel:  AVX+FMA - 1 threads (8 CPU cores detected)\r\n",
      "\r\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 8 cores!\r\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\r\n",
      "\r\n",
      "Reading alignment file alig_clustalo.fa ... Fasta format detected\r\n",
      "Alignment most likely contains protein sequences\r\n",
      "Alignment has 201 sequences with 434 columns, 340 distinct patterns\r\n",
      "220 parsimony-informative, 78 singleton sites, 136 constant sites\r\n",
      "                                                     Gap/Ambiguity  Composition  p-value\r\n",
      "   1  QL10p018382_mrna_CDS_2_Quercus_lobata                 69.82%    passed     98.82%\r\n",
      "   2  QL06p031114_mrna_CDS_2_Quercus_lobata                 70.05%    passed     99.91%\r\n",
      "   3  QL10p018398_mrna_CDS_1_Quercus_lobata                 69.82%    passed     99.65%\r\n",
      "   4  QL11p027894_mrna_CDS_1_Quercus_lobata                 70.05%    passed     99.10%\r\n",
      "   5  QL10p018395_mrna_CDS_1_Quercus_lobata                 70.05%    passed     38.33%\r\n",
      "   6  QL05p056248_mrna_Quercus_lobata                       75.35%    passed     60.87%\r\n",
      "   7  QL10p018379_mrna_CDS_1_Quercus_lobata                 73.73%    passed     75.47%\r\n",
      "   8  QL06p031123_mrna_CDS_1_Quercus_lobata                 70.05%    passed     98.06%\r\n",
      "   9  PNT58448_Populus_trichocarpa                          68.43%    passed     93.84%\r\n",
      "  10  PNT57551_Populus_trichocarpa                          70.05%    passed     16.20%\r\n",
      "  11  PNT57550_Populus_trichocarpa                          66.36%    passed     97.18%\r\n",
      "  12  PNT46426_Populus_trichocarpa                          67.97%    passed     92.64%\r\n",
      "  13  PNT20170_Populus_trichocarpa                          67.74%    passed     96.16%\r\n",
      "  14  PNT12323_Populus_trichocarpa                          90.09%    passed     11.90%\r\n",
      "  15  CDS_HanXRQr2_Chr12g0556091.1_Helianthus_annuus        64.29%    passed     64.61%\r\n",
      "  16  CDS_HanXRQr2_Chr15g0699451.1_Helianthus_annuus        66.59%    passed     40.23%\r\n",
      "  17  CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus        75.35%    passed     88.48%\r\n",
      "  18  CDS_HanXRQr2_Chr02g0065251.1_Helianthus_annuus        69.82%    passed     94.94%\r\n",
      "  19  CDS_HanXRQr2_Chr15g0708181.1_Helianthus_annuus        65.90%    failed      0.01%\r\n",
      "  20  CDS_HanXRQr2_Chr04g0158231.1_Helianthus_annuus        70.51%    passed     87.64%\r\n",
      "  21  CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus        69.82%    passed     95.29%\r\n",
      "  22  Bo8g101500.1_Brassica_oleracea                        70.74%    passed     92.16%\r\n",
      "  23  Bo7g119230.1_Brassica_oleracea                        70.51%    passed     90.87%\r\n",
      "  24  Bo2g088760.1_Brassica_oleracea                        57.83%    failed      0.28%\r\n",
      "  25  Bo1g002490.1_Brassica_oleracea                        69.82%    passed     92.32%\r\n",
      "  26  Bo2g091120.1_Brassica_oleracea                        64.29%    passed     67.71%\r\n",
      "  27  Bo4g147120.1_Brassica_oleracea                        71.43%    passed     92.67%\r\n",
      "  28  Bo6g083750.1_Brassica_oleracea                        68.43%    passed     80.67%\r\n",
      "  29  RHC11H1G0771.2.1.cds.1_Solanum_tuberosum_rh8903916    67.74%    failed      4.46%\r\n",
      "  30  RHC11H1G0775.2.1.cds.1_Solanum_tuberosum_rh8903916    69.82%    passed     69.54%\r\n",
      "  31  RHC04H1G2339.2.1.cds.1_Solanum_tuberosum_rh8903916    70.05%    passed     96.19%\r\n",
      "  32  KJB60013_Gossypium_raimondii                          70.51%    passed     90.69%\r\n",
      "  33  KJB57150_Gossypium_raimondii                          68.43%    passed     48.19%\r\n",
      "  34  KJB71292_Gossypium_raimondii                          71.43%    passed     99.89%\r\n",
      "  35  KZN08747_Daucus_carota                                75.35%    passed     93.99%\r\n",
      "  36  KZN08752_Daucus_carota                                67.28%    passed     83.77%\r\n",
      "  37  ERN08734_Amborella_trichopoda                         70.97%    passed     64.91%\r\n",
      "  38  MD10G1029400_Mdomestica_491_v1                        64.75%    passed     98.64%\r\n",
      "  39  MD15G1340900_Mdomestica_491_v1                        65.67%    passed     34.84%\r\n",
      "  40  MD01G1001000_Mdomestica_491_v1                        65.44%    passed     30.53%\r\n",
      "  41  HORVU7Hr1G102010.1_Hordeum_vulgare                    70.05%    passed     16.93%\r\n",
      "  42  HORVU4Hr1G055410.2_Hordeum_vulgare                    63.82%    passed     63.48%\r\n",
      "  43  HORVU4Hr1G055410.3_Hordeum_vulgare                    71.20%    passed     90.81%\r\n",
      "  44  HORVU7Hr1G101980.1_Hordeum_vulgare                    59.91%    failed      1.94%\r\n",
      "  45  HORVU4Hr1G020010.2_Hordeum_vulgare                    74.42%    passed      9.96%\r\n",
      "  46  HORVU4Hr1G020010.1_Hordeum_vulgare                    67.97%    failed      2.89%\r\n",
      "  47  Aco010349.1.mrna1_Ananas_comosus                      34.10%    failed      0.00%\r\n",
      "  48  Aco016990.1.mrna1.cds1_Ananas_comosus                 71.20%    passed     78.67%\r\n",
      "  49  Aco029478.1.mrna1.cds1_Ananas_comosus                 63.59%    failed      2.92%\r\n",
      "  50  Aco003193.1.mrna1.cds1_Ananas_comosus                 63.59%    failed      2.92%\r\n",
      "  51  Aco019928.1.mrna1.cds1_Ananas_comosus                 81.34%    passed     28.40%\r\n",
      "  52  Aco024757.1.mrna1.cds1_Ananas_comosus                 77.88%    passed     62.91%\r\n",
      "  53  Aco024977.1.mrna1.cds1_Ananas_comosus                 76.96%    failed      0.01%\r\n",
      "  54  Aco018685.1.mrna1.cds1_Ananas_comosus                 67.51%    passed     35.13%\r\n",
      "  55  Aco031263.1.mrna1.cds1_Ananas_comosus                 88.48%    passed     73.37%\r\n",
      "  56  PEQU_28350.1_Phalaenopsis_equestris                   69.59%    passed     27.81%\r\n",
      "  57  ONH90092_Prunus_persica                               71.89%    passed     60.60%\r\n",
      "  58  ONH90042_Prunus_persica                               64.98%    passed     89.78%\r\n",
      "  59  ONH90434_Prunus_persica                               71.43%    passed     56.63%\r\n",
      "  60  ONH90087_Prunus_persica                               70.74%    passed     98.67%\r\n",
      "  61  ONH90046_Prunus_persica                               64.98%    passed     89.39%\r\n",
      "  62  ONH89505_Prunus_persica                               67.28%    passed     84.87%\r\n",
      "  63  ONI01825_Prunus_persica                               60.37%    failed      2.36%\r\n",
      "  64  Os11t0168000-01_Oryza_sativa                          66.82%    failed      1.98%\r\n",
      "  65  Os03t0296600-01_Oryza_sativa                          65.21%    passed     15.54%\r\n",
      "  66  EES16714_Sorghum_bicolor                              59.22%    failed      0.42%\r\n",
      "  67  EER95019_Sorghum_bicolor                              60.60%    failed      3.50%\r\n",
      "  68  EOX92363_Theobroma_cacao                              69.35%    passed     92.38%\r\n",
      "  69  EOY04380_Theobroma_cacao                              68.66%    passed     41.66%\r\n",
      "  70  EOY04381_Theobroma_cacao                              69.12%    passed     83.93%\r\n",
      "  71  EOY18917_Theobroma_cacao                              69.12%    passed     29.40%\r\n",
      "  72  EOY16769_Theobroma_cacao                              67.74%    passed     70.05%\r\n",
      "  73  EOY27508_Theobroma_cacao                              69.35%    passed     99.27%\r\n",
      "  74  Ma08_p15310.1_Musa_acuminata                          67.74%    passed     60.06%\r\n",
      "  75  Ma06_p27500.1_Musa_acuminata                          63.13%    passed     72.42%\r\n",
      "  76  Solyc04g071640.1.1.1_Solanum_lycopersicum             70.28%    passed     86.41%\r\n",
      "  77  Solyc11g005500.1.1.1_Solanum_lycopersicum             70.05%    passed     58.08%\r\n",
      "  78  Solyc11g005540.1.1.1_Solanum_lycopersicum             68.66%    passed      6.91%\r\n",
      "  79  itb06g24100.t1_Ipomoea_triloba                        70.97%    passed     55.96%\r\n",
      "  80  itb04g21400.t1_Ipomoea_triloba                        65.90%    passed     70.02%\r\n",
      "  81  itb05g22120.t1_Ipomoea_triloba                        59.91%    passed     48.26%\r\n",
      "  82  OIW19998_Lupinus_angustifolius                        69.12%    passed     95.46%\r\n",
      "  83  OIW19997_Lupinus_angustifolius                        69.35%    failed      0.86%\r\n",
      "  84  OIW20030_Lupinus_angustifolius                        73.73%    passed     98.39%\r\n",
      "  85  Csa16g040920.1.cds1_Camelina_sativa                   65.21%    passed     62.21%\r\n",
      "  86  Csa07g048320.1.cds1_Camelina_sativa                   65.21%    passed     75.02%\r\n",
      "  87  Csa12g002210.1.cds1_Camelina_sativa                   71.43%    passed     55.75%\r\n",
      "  88  Csa07g048380.1.cds1_Camelina_sativa                   87.79%    failed      2.29%\r\n",
      "  89  Csa10g002090.1.cds1_Camelina_sativa                   71.43%    passed     52.21%\r\n",
      "  90  Csa16g043520.1.cds1_Camelina_sativa                   71.20%    passed     71.21%\r\n",
      "  91  Csa07g051890.1.cds1_Camelina_sativa                   70.74%    passed     64.35%\r\n",
      "  92  Csa09g081500.1.cds1_Camelina_sativa                   64.98%    passed     75.54%\r\n",
      "  93  Csa09g086140.1.cds1_Camelina_sativa                   71.20%    passed     81.27%\r\n",
      "  94  Csa11g002350.1.cds1_Camelina_sativa                   71.43%    passed     45.62%\r\n",
      "  95  Csa09g086150.1.cds1_Camelina_sativa                   70.97%    passed     75.77%\r\n",
      "  96  Csa27089s010.1.cds1_Camelina_sativa                   86.64%    passed     30.99%\r\n",
      "  97  Csa16g043530.1_Camelina_sativa                        67.51%    failed      1.56%\r\n",
      "  98  Csa07g051880.1_Camelina_sativa                        77.88%    passed     73.86%\r\n",
      "  99  FCD_00011687-RA_cds_Ficus_carica                      72.81%    passed     94.94%\r\n",
      " 100  FCD_00028685-RA_cds_Ficus_carica                      68.43%    passed     92.49%\r\n",
      " 101  FCD_00030006-RA_cds_Ficus_carica                      68.43%    passed     91.12%\r\n",
      " 102  FCD_00011685-RA_cds_Ficus_carica                      64.98%    passed     62.68%\r\n",
      " 103  FvH4_2g18700.t1_Fvesca_677_v4                         65.90%    passed     30.46%\r\n",
      " 104  FvH4_2g18910.t1_Fvesca_677_v4                         66.82%    passed     91.87%\r\n",
      " 105  FvH4_2g13400.t1_Fvesca_677_v4                         64.29%    passed     88.15%\r\n",
      " 106  FvH4_2g13390.t1_Fvesca_677_v4                         64.52%    passed     85.98%\r\n",
      " 107  FvH4_2g18870.t1_Fvesca_677_v4                         68.20%    passed     51.33%\r\n",
      " 108  FvH4_4g25660.t1_Fvesca_677_v4                         69.35%    passed     60.19%\r\n",
      " 109  FvH4_1g23830.t1_Fvesca_677_v4                         67.28%    passed     37.76%\r\n",
      " 110  FvH4_5g02500.t1_Fvesca_677_v4                         66.13%    passed     86.12%\r\n",
      " 111  RZC48302_Papaver_somniferum                           63.36%    passed     88.88%\r\n",
      " 112  RZC49191_Papaver_somniferum                           66.36%    passed     84.56%\r\n",
      " 113  RZC49225_Papaver_somniferum                           64.06%    passed     66.45%\r\n",
      " 114  RZC55249_Papaver_somniferum                           64.06%    passed     67.08%\r\n",
      " 115  RZC55281_Papaver_somniferum                           64.98%    passed     70.13%\r\n",
      " 116  RZC57082_Papaver_somniferum                           67.51%    passed     46.70%\r\n",
      " 117  RZC57075_Papaver_somniferum                           81.11%    passed     72.45%\r\n",
      " 118  RZC57081_Papaver_somniferum                           63.59%    passed     97.93%\r\n",
      " 119  RZC62100_Papaver_somniferum                           65.21%    passed     99.20%\r\n",
      " 120  RZC62099_Papaver_somniferum                           64.98%    passed     99.83%\r\n",
      " 121  RZC68710_Papaver_somniferum                           69.35%    passed     99.09%\r\n",
      " 122  RZC68605_Papaver_somniferum                           74.88%    failed      0.03%\r\n",
      " 123  RZC68602_Papaver_somniferum                           65.21%    passed     99.88%\r\n",
      " 124  PRQ49244_Rosa_chinensis                               76.50%    passed     81.65%\r\n",
      " 125  PRQ49241_Rosa_chinensis                               68.89%    passed     56.88%\r\n",
      " 126  PRQ25531_Rosa_chinensis                               66.13%    passed     70.49%\r\n",
      " 127  PRQ24668_Rosa_chinensis                               57.14%    passed     10.97%\r\n",
      " 128  PRQ25532_Rosa_chinensis                               66.36%    passed     72.64%\r\n",
      " 129  PRQ17678_Rosa_chinensis                               69.12%    passed     99.48%\r\n",
      " 130  Vradi07g02140.1_Vigna_radiata                         69.35%    passed     50.99%\r\n",
      " 131  Vradi01g10860.1_Vigna_radiata                         73.27%    passed     40.05%\r\n",
      " 132  OIT31467_Nicotiana_attenuata                          70.97%    passed     55.42%\r\n",
      " 133  ESW31795_Phaseolus_vulgaris                           75.12%    passed     57.80%\r\n",
      " 134  ESW18393_Phaseolus_vulgaris                           70.28%    passed     71.71%\r\n",
      " 135  ESW28268_Phaseolus_vulgaris                           68.20%    passed     97.52%\r\n",
      " 136  KEH21230_Medicago_truncatula                          67.51%    passed     25.11%\r\n",
      " 137  AES70181_Medicago_truncatula                          66.59%    passed     72.41%\r\n",
      " 138  AES70191_Medicago_truncatula                          70.51%    passed     25.05%\r\n",
      " 139  OAY36032_Manihot_esculenta                            68.89%    passed     98.02%\r\n",
      " 140  OAY36033_Manihot_esculenta                            65.21%    passed     67.80%\r\n",
      " 141  KRH63270_Glycine_max                                  68.20%    passed     72.51%\r\n",
      " 142  KRH59512_Glycine_max                                  72.35%    passed     48.06%\r\n",
      " 143  KRH54635_Glycine_max                                  68.43%    passed     83.06%\r\n",
      " 144  KRH43377_Glycine_max                                  72.35%    passed     79.95%\r\n",
      " 145  KRH39377_Glycine_max                                  69.35%    passed     50.06%\r\n",
      " 146  KRH35123_Glycine_max                                  84.10%    passed     72.68%\r\n",
      " 147  KRG98863_Glycine_max                                  74.65%    passed     50.27%\r\n",
      " 148  KRG91676_Glycine_max                                  67.51%    passed     76.84%\r\n",
      " 149  KMT04640_Beta_vulgaris                                67.28%    passed     80.11%\r\n",
      " 150  KMT16083_Beta_vulgaris                                73.50%    passed     36.74%\r\n",
      " 151  DRNTG_09031.1.cds1_Dioscorea_rotundata                79.95%    passed     61.92%\r\n",
      " 152  FtPinG0007730800.01.T01_Fagopyrum_tataricum           76.73%    passed     60.82%\r\n",
      " 153  KGN64838_Cucumis_sativus                              67.28%    passed     76.97%\r\n",
      " 154  KGN64837_Cucumis_sativus                              73.27%    passed     55.64%\r\n",
      " 155  KGN46138_Cucumis_sativus                              69.35%    passed     80.13%\r\n",
      " 156  cds.evm.model.01.1369_Cannabis_sativa_female          68.66%    passed     31.77%\r\n",
      " 157  cds.evm.model.04.1284_Cannabis_sativa_female          62.90%    passed     78.67%\r\n",
      " 158  cds.Jr15_01190_p1_Juglans_regia                       64.06%    passed     71.70%\r\n",
      " 159  VIT_09s0002g01570.t01_Vitis_vinifera                  70.51%    passed     63.84%\r\n",
      " 160  VIT_07s0031g02070.t01_Vitis_vinifera                  68.89%    passed     96.84%\r\n",
      " 161  OE9A117531C1_Olea_europaea                            68.89%    passed     60.28%\r\n",
      " 162  NC1G0017660.1_cds_Nymphaea_colorata                   71.20%    passed     57.26%\r\n",
      " 163  NC11G0118510.1_cds_Nymphaea_colorata                  70.51%    passed     14.65%\r\n",
      " 164  NC4G0237690.1_cds_Nymphaea_colorata                   86.41%    passed     35.45%\r\n",
      " 165  ESR47720_Citrus_clementina                            67.05%    passed     99.99%\r\n",
      " 166  Zm00001eb396810_P001_Zea_mays                         63.13%    passed     13.90%\r\n",
      " 167  Kaladp0040s0523.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   69.59%    passed     90.16%\r\n",
      " 168  Kaladp0026s0134.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   67.97%    passed     94.81%\r\n",
      " 169  Kaladp0515s0048.1.v1.1_Kalanchoe_fedtschenkoi         74.19%    failed      0.00%\r\n",
      " 170  Kaladp0033s0187.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi   69.12%    passed     85.20%\r\n",
      " 171  ESQ53189_Eutrema_salsugineum                          70.51%    passed     62.97%\r\n",
      " 172  ESQ46086_Eutrema_salsugineum                          77.19%    passed     48.80%\r\n",
      " 173  ESQ46085_Eutrema_salsugineum                          71.20%    passed     83.77%\r\n",
      " 174  ESQ27444_Eutrema_salsugineum                          63.36%    passed     44.08%\r\n",
      " 175  SIN_1014976.t.cds1_Sesamum_indicum                    70.05%    passed     62.90%\r\n",
      " 176  SIN_1019387.t.cds1_Sesamum_indicum                    68.89%    passed     77.92%\r\n",
      " 177  CDS_MD10G0023400.1_Malus_domestica_golden             64.75%    passed     98.64%\r\n",
      " 178  CDS_MD01G0004200.1_Malus_domestica_golden             65.44%    passed     30.53%\r\n",
      " 179  TraesCS4D02G109400.1.cds1_Triticum_aestivum           64.52%    failed      0.66%\r\n",
      " 180  TraesCS7D02G440400.1.cds1_Triticum_aestivum           61.75%    passed      9.31%\r\n",
      " 181  TraesCS7A02G451500.1.cds1_Triticum_aestivum           62.21%    passed     13.08%\r\n",
      " 182  TraesCS7A02G451600.1.cds1_Triticum_aestivum           65.21%    passed      7.15%\r\n",
      " 183  TraesCS7B02G351400.1.cds1_Triticum_aestivum           65.44%    passed     10.22%\r\n",
      " 184  TraesCS7B02G351300.1.cds1_Triticum_aestivum           61.29%    passed     30.04%\r\n",
      " 185  TraesCS4B02G190500.1.cds1_Triticum_aestivum           61.98%    passed     20.82%\r\n",
      " 186  TraesCS7D02G440500.1.cds1_Triticum_aestivum           65.21%    failed      4.87%\r\n",
      " 187  TraesCS4B02G111900.1_Triticum_aestivum                66.59%    failed      0.01%\r\n",
      " 188  TraesCS4A02G208200.1.cds1_Triticum_aestivum           64.52%    failed      0.52%\r\n",
      " 189  TraesCS4A02G113600.1.cds1_Triticum_aestivum           63.82%    passed     22.11%\r\n",
      " 190  AT4G39340.1_Arabidopsis_thaliana                      70.74%    passed     69.58%\r\n",
      " 191  AT2G21750.1_Arabidopsis_thaliana                      71.20%    passed     75.17%\r\n",
      " 192  AT1G76750.1_Arabidopsis_thaliana                      63.59%    passed     65.49%\r\n",
      " 193  AT2G21740.1_Arabidopsis_thaliana                      71.20%    passed     87.23%\r\n",
      " 194  cds-PLY89992.1_Lactuca_sativa                         66.13%    passed     13.27%\r\n",
      " 195  cds-PLY89774.1_Lactuca_sativa                         69.82%    passed     32.95%\r\n",
      " 196  CDP12837_Coffea_canephora                             64.52%    passed     49.67%\r\n",
      " 197  Bradi2g28070.1.p_Bdistachyon_314_v3                   66.82%    passed     28.63%\r\n",
      " 198  Bradi4g24640.1.p_Bdistachyon_314_v3                   63.13%    failed      4.17%\r\n",
      " 199  Bradi1g65190.1.p_Bdistachyon_314_v3                   60.14%    failed      0.94%\r\n",
      " 200  PSS26677_Actinidia_chinensis                          68.43%    passed     98.71%\r\n",
      " 201  jg50396.t1_Fagopyrum_esculentum                       72.58%    passed     77.37%\r\n",
      "WARNING: 200 sequences contain more than 50% gaps/ambiguity\r\n",
      "****  TOTAL                                                 68.78%  24 sequences failed composition chi2 test (p-value<5%; df=19)\r\n",
      "NOTE: CDS_MD10G0023400.1_Malus_domestica_golden is identical to MD10G1029400_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD01G0004200.1_Malus_domestica_golden is identical to MD01G1001000_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: Aco003193.1.mrna1.cds1_Ananas_comosus is identical to Aco029478.1.mrna1.cds1_Ananas_comosus but kept for subsequent analysis\r\n",
      "\r\n",
      "NOTE: Restoring information from model checkpoint file clustalo_not_trimmed_iqtree_bootstrap.model.gz\r\n",
      "\r\n",
      "CHECKPOINT: Initial tree restored\r\n",
      "Perform fast likelihood tree search using LG+I+G model...\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHECKPOINT: Tree restored, LogL: -47290.310\n",
      "NOTE: ModelFinder requires 113 MB RAM!\n",
      "ModelFinder will test up to 546 protein models (sample size: 434) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            32184.915    399 65167.829    74556.064    66792.974\n",
      "  2  LG+I          32184.898    400 65169.796    74891.008    66799.014\n",
      "  3  LG+G4         30785.709    400 62371.417    72092.630    64000.635\n",
      "  4  LG+I+G4       30785.723    401 62373.446    72448.571    64006.737\n",
      "  5  LG+R2         30950.591    401 62703.183    72778.308    64336.474\n",
      "  6  LG+R3         30800.223    403 62406.446    73260.580    64047.883\n",
      "  7  LG+R4         30741.471    405 62292.943    74037.943    63942.526\n",
      "  8  LG+R5         30721.892    407 62257.783    75031.322    63915.512\n",
      "  9  LG+R6         30711.126    409 62240.252    76214.418    63906.127\n",
      " 10  LG+R7         30710.942    411 62243.885    77637.703    63917.906\n",
      " 21  LG+F+R5       30758.684    426 62369.368    114341.368   64104.485\n",
      " 22  LG+F+R6       30748.592    428 62353.183    135797.983   64096.446\n",
      " 34  WAG+R5        30523.033    407 61860.067    74633.605    63517.796\n",
      " 35  WAG+R6        30519.084    409 61856.167    75830.334    63522.043\n",
      " 47  WAG+F+R5      30481.769    426 61815.537    113787.537   63550.654\n",
      " 48  WAG+F+R6      30477.749    428 61811.497    135256.297   63554.761\n",
      " 60  JTT+R5        30262.255    407 61338.511    74112.049    62996.240\n",
      " 61  JTT+R6        30259.096    409 61336.192    75310.359    63002.068\n",
      " 73  JTT+F+R5      30366.910    426 61585.819    113557.819   63320.936\n",
      " 74  JTT+F+R6      30362.954    428 61581.908    135026.708   63325.171\n",
      " 86  JTTDCMut+R5   30271.616    407 61357.231    74130.770    63014.961\n",
      " 87  JTTDCMut+R6   30268.273    409 61354.546    75328.713    63020.422\n",
      " 99  JTTDCMut+F+R5 30369.686    426 61591.372    113563.372   63326.489\n",
      "100  JTTDCMut+F+R6 30365.391    428 61586.782    135031.582   63330.045\n",
      "112  DCMut+R5      30710.755    407 62235.510    75009.049    63893.240\n",
      "113  DCMut+R6      30707.943    409 62233.886    76208.053    63899.761\n",
      "125  DCMut+F+R5    30621.951    426 62095.903    114067.903   63831.020\n",
      "126  DCMut+F+R6    30618.137    428 62092.273    135537.073   63835.536\n",
      "138  VT+R5         30530.178    407 61874.356    74647.895    63532.086\n",
      "139  VT+R6         30528.516    409 61875.031    75849.198    63540.907\n",
      "151  VT+F+R5       30567.549    426 61987.098    113959.098   63722.215\n",
      "152  VT+F+R6       30565.186    428 61986.371    135431.171   63729.634\n",
      "164  PMB+R5        30845.481    407 62504.962    75278.501    64162.691\n",
      "165  PMB+R6        30843.831    409 62505.661    76479.828    64171.536\n",
      "177  PMB+F+R5      30795.515    426 62443.031    114415.031   64178.148\n",
      "178  PMB+F+R6      30793.326    428 62442.653    135887.453   64185.916\n",
      "190  Blosum62+R5   30996.928    407 62807.856    75581.394    64465.585\n",
      "191  Blosum62+R6   30993.710    409 62805.420    76779.587    64471.296\n",
      "203  Blosum62+F+R5 30865.464    426 62582.928    114554.928   64318.045\n",
      "204  Blosum62+F+R6 30862.304    428 62580.608    136025.408   64323.871\n",
      "216  Dayhoff+R5    30710.597    407 62235.194    75008.732    63892.923\n",
      "217  Dayhoff+R6    30707.748    409 62233.496    76207.663    63899.372\n",
      "229  Dayhoff+F+R5  30622.584    426 62097.168    114069.168   63832.285\n",
      "230  Dayhoff+F+R6  30618.711    428 62093.421    135538.221   63836.684\n",
      "242  mtREV+R5      31421.062    407 63656.123    76429.662    65313.852\n",
      "243  mtREV+R6      31411.861    409 63641.721    77615.888    65307.597\n",
      "255  mtREV+F+R5    30811.257    426 62474.513    114446.513   64209.630\n",
      "256  mtREV+F+R6    30799.309    428 62454.617    135899.417   64197.880\n",
      "268  mtART+R5      32487.854    407 65789.708    78563.246    67447.437\n",
      "269  mtART+R6      32428.575    409 65675.151    79649.317    67341.026\n",
      "281  mtART+F+R5    31808.401    426 64468.802    116440.802   66203.919\n",
      "282  mtART+F+R6    31775.854    428 64407.709    137852.509   66150.972\n",
      "294  mtZOA+R5      31598.453    407 64010.906    76784.445    65668.635\n",
      "295  mtZOA+R6      31552.520    409 63923.041    77897.207    65588.916\n",
      "307  mtZOA+F+R5    31310.962    426 63473.924    115445.924   65209.041\n",
      "308  mtZOA+F+R6    31274.819    428 63405.638    136850.438   65148.902\n",
      "320  mtMet+R5      31532.503    407 63879.005    76652.544    65536.734\n",
      "321  mtMet+R6      31523.651    409 63865.302    77839.468    65531.177\n",
      "333  mtMet+F+R5    30750.674    426 62353.348    114325.348   64088.465\n",
      "334  mtMet+F+R6    30736.901    428 62329.801    135774.601   64073.064\n",
      "346  mtVer+R5      31364.842    407 63543.684    76317.223    65201.413\n",
      "347  mtVer+R6      31352.278    409 63522.556    77496.722    65188.431\n",
      "359  mtVer+F+R5    30804.569    426 62461.138    114433.138   64196.255\n",
      "360  mtVer+F+R6    30789.158    428 62434.317    135879.117   64177.580\n",
      "372  mtInv+R5      31988.822    407 64791.644    77565.183    66449.373\n",
      "373  mtInv+R6      31981.464    409 64780.929    78755.095    66446.804\n",
      "385  mtInv+F+R5    30755.289    426 62362.578    114334.578   64097.695\n",
      "386  mtInv+F+R6    30743.264    428 62342.527    135787.327   64085.790\n",
      "398  mtMAM+R5      31833.021    407 64480.042    77253.581    66137.772\n",
      "399  mtMAM+R6      31813.799    409 64445.599    78419.765    66111.474\n",
      "411  mtMAM+F+R5    31101.568    426 63055.136    115027.136   64790.253\n",
      "412  mtMAM+F+R6    31089.489    428 63034.979    136479.779   64778.242\n",
      "424  HIVb+R5       30918.824    407 62651.648    75425.186    64309.377\n",
      "425  HIVb+R6       30911.598    409 62641.195    76615.362    64307.071\n",
      "437  HIVb+F+R5     30672.203    426 62196.405    114168.405   63931.522\n",
      "438  HIVb+F+R6     30662.382    428 62180.764    135625.564   63924.027\n",
      "450  HIVw+R5       31733.448    407 64280.896    77054.434    65938.625\n",
      "451  HIVw+R6       31728.640    409 64275.281    78249.447    65941.156\n",
      "463  HIVw+F+R5     31067.519    426 62987.037    114959.037   64722.154\n",
      "464  HIVw+F+R6     31063.404    428 62982.809    136427.609   64726.072\n",
      "476  FLU+R5        30862.475    407 62538.950    75312.488    64196.679\n",
      "477  FLU+R6        30853.149    409 62524.297    76498.464    64190.172\n",
      "489  FLU+F+R5      30496.951    426 61845.902    113817.902   63581.019\n",
      "490  FLU+F+R6      30487.773    428 61831.546    135276.346   63574.809\n",
      "502  rtREV+R5      31071.358    407 62956.716    75730.255    64614.445\n",
      "503  rtREV+R6      31062.809    409 62943.618    76917.785    64609.493\n",
      "515  rtREV+F+R5    30980.893    426 62813.786    114785.786   64548.903\n",
      "516  rtREV+F+R6    30968.349    428 62792.698    136237.498   64535.961\n",
      "528  cpREV+R5      30670.180    407 62154.360    74927.899    63812.089\n",
      "529  cpREV+R6      30667.289    409 62152.579    76126.745    63818.454\n",
      "541  cpREV+F+R5    30706.894    426 62265.788    114237.788   64000.905\n",
      "542  cpREV+F+R6    30693.202    428 62242.403    135687.203   63985.666\n",
      "Akaike Information Criterion:           JTT+R6\n",
      "Corrected Akaike Information Criterion: LG+G4\n",
      "Bayesian Information Criterion:         JTT+R5\n",
      "Best-fit model: JTT+R5 chosen according to BIC\n",
      "\n",
      "All model information printed to clustalo_not_trimmed_iqtree_bootstrap.model.gz\n",
      "CPU time for ModelFinder: 0.076 seconds (0h:0m:0s)\n",
      "Wall-clock time for ModelFinder: 0.088 seconds (0h:0m:0s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 927732)...\n",
      "\n",
      "NOTE: 57 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -47111.742\n",
      "2. Current log-likelihood: -40240.438\n",
      "3. Current log-likelihood: -40175.782\n",
      "4. Current log-likelihood: -40171.772\n",
      "5. Current log-likelihood: -40169.216\n",
      "6. Current log-likelihood: -40167.450\n",
      "7. Current log-likelihood: -40166.265\n",
      "8. Current log-likelihood: -40165.311\n",
      "9. Current log-likelihood: -40164.597\n",
      "10. Current log-likelihood: -40163.954\n",
      "11. Current log-likelihood: -40163.381\n",
      "12. Current log-likelihood: -40162.874\n",
      "13. Current log-likelihood: -40162.419\n",
      "14. Current log-likelihood: -40162.021\n",
      "15. Current log-likelihood: -40161.667\n",
      "16. Current log-likelihood: -40161.348\n",
      "17. Current log-likelihood: -40161.065\n",
      "18. Current log-likelihood: -40160.901\n",
      "19. Current log-likelihood: -40160.659\n",
      "20. Current log-likelihood: -40160.525\n",
      "21. Current log-likelihood: -40160.392\n",
      "22. Current log-likelihood: -40160.270\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23. Current log-likelihood: -40160.159\n",
      "24. Current log-likelihood: -40160.056\n",
      "Optimal log-likelihood: -40159.903\n",
      "Site proportion and rates:  (0.109,0.088) (0.149,0.318) (0.168,0.670) (0.292,1.125) (0.283,1.776)\n",
      "Parameters optimization took 24 rounds (84.727 sec)\n",
      "Computing ML distances based on estimated model parameters... 5.243 sec\n",
      "WARNING: Some pairwise ML distances are too long (saturated)\n",
      "Computing BIONJ tree...\n",
      "0.017 seconds\n",
      "Log-likelihood of BIONJ tree: -30664.524\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 10.396 second\n",
      "Computing log-likelihood of 98 initial trees ... 36.176 seconds\n",
      "Current best score: -30664.524\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -30420.803\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -30404.459\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 5: -30390.904\n",
      "Iteration 10 / LogL: -30416.270 / Time: 0h:4m:30s\n",
      "Iteration 20 / LogL: -30414.353 / Time: 0h:6m:28s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -30390.904 / CPU time: 296.160\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -30393.940 / Time: 0h:8m:18s (0h:21m:30s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 36: -30387.369\n",
      "Iteration 40 / LogL: -30396.814 / Time: 0h:10m:14s (0h:25m:11s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 50: -30386.630\n",
      "Iteration 50 / LogL: -30386.630 / Time: 0h:12m:8s (0h:24m:47s left)\n",
      "Log-likelihood cutoff on original alignment: -30454.618\n",
      "Iteration 60 / LogL: -30387.483 / Time: 0h:13m:57s (0h:21m:18s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 62: -30382.678\n",
      "Iteration 70 / LogL: -30387.750 / Time: 0h:15m:46s (0h:21m:1s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 72: -30382.393\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 77: -30380.845\n",
      "Iteration 80 / LogL: -30382.380 / Time: 0h:17m:42s (0h:21m:45s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 84: -30377.155\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 86: -30376.852\n",
      "Iteration 90 / LogL: -30384.450 / Time: 0h:19m:31s (0h:21m:3s left)\n",
      "Iteration 100 / LogL: -30385.385 / Time: 0h:21m:37s (0h:18m:47s left)\n",
      "Log-likelihood cutoff on original alignment: -30454.618\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.985\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 110 / LogL: -30404.019 / Time: 0h:23m:35s (0h:19m:29s left)\n",
      "Iteration 120 / LogL: -30377.467 / Time: 0h:25m:16s (0h:16m:59s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 122: -30374.850\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 126: -30372.717\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 127: -30372.578\n",
      "Iteration 130 / LogL: -30439.388 / Time: 0h:27m:23s (0h:20m:35s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 136: -30372.455\n",
      "Iteration 140 / LogL: -30393.702 / Time: 0h:29m:24s (0h:20m:18s left)\n",
      "Iteration 150 / LogL: -30372.600 / Time: 0h:31m:12s (0h:18m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -30440.886\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 155: -30371.453\n",
      "Iteration 160 / LogL: -30373.012 / Time: 0h:33m:6s (0h:19m:46s left)\n",
      "Iteration 170 / LogL: -30405.853 / Time: 0h:34m:58s (0h:17m:35s left)\n",
      "Iteration 180 / LogL: -30383.018 / Time: 0h:36m:51s (0h:15m:26s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 187: -30371.208\n",
      "UPDATE BEST LOG-LIKELIHOOD: -30371.203\n",
      "Iteration 190 / LogL: -30376.623 / Time: 0h:38m:54s (0h:19m:57s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 194: -30371.065\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 197: -30370.881\n",
      "Iteration 200 / LogL: -30372.548 / Time: 0h:41m:19s (0h:20m:8s left)\n",
      "Log-likelihood cutoff on original alignment: -30433.632\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.986\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 210 / LogL: -30378.271 / Time: 0h:43m:30s (0h:18m:44s left)\n",
      "Iteration 220 / LogL: -30378.888 / Time: 0h:45m:29s (0h:16m:37s left)\n",
      "Iteration 230 / LogL: -30376.532 / Time: 0h:47m:47s (0h:14m:36s left)\n",
      "Iteration 240 / LogL: -30378.528 / Time: 0h:49m:34s (0h:12m:26s left)\n",
      "Iteration 250 / LogL: -30372.627 / Time: 0h:51m:37s (0h:10m:21s left)\n",
      "Log-likelihood cutoff on original alignment: -30433.124\n",
      "Iteration 260 / LogL: -30411.662 / Time: 0h:53m:34s (0h:8m:16s left)\n",
      "Iteration 270 / LogL: -30371.433 / Time: 0h:55m:30s (0h:6m:11s left)\n",
      "Iteration 280 / LogL: -30381.753 / Time: 0h:57m:29s (0h:4m:7s left)\n",
      "Iteration 290 / LogL: -30374.497 / Time: 0h:59m:16s (0h:2m:3s left)\n",
      "Iteration 300 / LogL: -30372.857 / Time: 1h:1m:15s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -30433.124\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.993\n",
      "TREE SEARCH COMPLETED AFTER 300 ITERATIONS / Time: 1h:1m:15s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -30370.881\n",
      "Optimal log-likelihood: -30370.877\n",
      "Site proportion and rates:  (0.092,0.102) (0.110,0.267) (0.135,0.504) (0.219,0.915) (0.444,1.562)\n",
      "Parameters optimization took 1 rounds (3.043 sec)\n",
      "BEST SCORE FOUND : -30370.877\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 1000 replicates...\n",
      "Testing tree branches by aBayes parametric test...\n",
      "4.452 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file clustalo_not_trimmed_iqtree_bootstrap.splits.nex\n",
      "Total tree length: 90.449\n",
      "\n",
      "Total number of iterations: 300\n",
      "CPU time used for tree search: 3576.898 sec (0h:59m:36s)\n",
      "Wall-clock time used for tree search: 3582.819 sec (0h:59m:42s)\n",
      "Total CPU time used: 3677.621 sec (1h:1m:17s)\n",
      "Total wall-clock time used: 3683.708 sec (1h:1m:23s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file clustalo_not_trimmed_iqtree_bootstrap.splits.nex...\n",
      "201 taxa and 1469 splits.\n",
      "Consensus tree written to clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "Reading input trees file clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "Log-likelihood of consensus tree: -30371.952\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                clustalo_not_trimmed_iqtree_bootstrap.iqtree\n",
      "  Maximum-likelihood tree:       clustalo_not_trimmed_iqtree_bootstrap.treefile\n",
      "  Likelihood distances:          clustalo_not_trimmed_iqtree_bootstrap.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          clustalo_not_trimmed_iqtree_bootstrap.splits.nex\n",
      "  Consensus tree:                clustalo_not_trimmed_iqtree_bootstrap.contree\n",
      "  Screen log file:               clustalo_not_trimmed_iqtree_bootstrap.log\n",
      "\n",
      "Date and Time: Sat May 21 16:55:18 2022\n"
     ]
    }
   ],
   "source": [
    "!iqtree -m MFP -s alig_clustalo.fa --prefix clustalo_not_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ccfe76a",
   "metadata": {},
   "source": [
    "## Trimming\n",
    "\n",
    "We also tried trimming the alignment with trimAl v1.4.rev15. We tried `gappyout` and `strict` trimming (`strict-plus` generated same result as `strict`).\n",
    "\n",
    "**Gappyout**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66c0f15b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!trimal -in alig_clustalo.fa -out gappyout_trimmed_alig_clustalo.fa -gappyout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "96fc03bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.3 for Linux 64-bit built Dec 20 2020\r\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\r\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\r\n",
      "\r\n",
      "Host:    AnnaToiThink (AVX512, FMA3, 31 GB RAM)\r\n",
      "Command: iqtree -m MFP -s gappyout_trimmed_alig_clustalo.fa --prefix clustalo_gappyout_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes\r\n",
      "Seed:    719585 (Using SPRNG - Scalable Parallel Random Number Generator)\r\n",
      "Time:    Sat May 21 13:02:15 2022\r\n",
      "Kernel:  AVX+FMA - 1 threads (8 CPU cores detected)\r\n",
      "\r\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 8 cores!\r\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\r\n",
      "\r\n",
      "Reading alignment file gappyout_trimmed_alig_clustalo.fa ... Fasta format detected\r\n",
      "Alignment most likely contains protein sequences\r\n",
      "Alignment has 201 sequences with 62 columns, 62 distinct patterns\r\n",
      "57 parsimony-informative, 5 singleton sites, 0 constant sites\r\n",
      "                                                     Gap/Ambiguity  Composition  p-value\r\n",
      "   1  QL10p018382_mrna_CDS_2_Quercus_lobata                  0.00%    passed     89.09%\r\n",
      "   2  QL06p031114_mrna_CDS_2_Quercus_lobata                  0.00%    passed     99.96%\r\n",
      "   3  QL10p018398_mrna_CDS_1_Quercus_lobata                  0.00%    passed     99.79%\r\n",
      "   4  QL11p027894_mrna_CDS_1_Quercus_lobata                  0.00%    passed     98.45%\r\n",
      "   5  QL10p018395_mrna_CDS_1_Quercus_lobata                  0.00%    passed     94.92%\r\n",
      "   6  QL05p056248_mrna_Quercus_lobata                       33.87%    passed      7.18%\r\n",
      "   7  QL10p018379_mrna_CDS_1_Quercus_lobata                  0.00%    passed     89.73%\r\n",
      "   8  QL06p031123_mrna_CDS_1_Quercus_lobata                  0.00%    passed     99.91%\r\n",
      "   9  PNT58448_Populus_trichocarpa                           0.00%    passed     96.72%\r\n",
      "  10  PNT57551_Populus_trichocarpa                           0.00%    passed     49.05%\r\n",
      "  11  PNT57550_Populus_trichocarpa                           0.00%    passed     98.35%\r\n",
      "  12  PNT46426_Populus_trichocarpa                           0.00%    passed     87.89%\r\n",
      "  13  PNT20170_Populus_trichocarpa                           0.00%    passed     76.67%\r\n",
      "  14  PNT12323_Populus_trichocarpa                          74.19%    passed     85.90%\r\n",
      "  15  CDS_HanXRQr2_Chr12g0556091.1_Helianthus_annuus         0.00%    passed     96.72%\r\n",
      "  16  CDS_HanXRQr2_Chr15g0699451.1_Helianthus_annuus         0.00%    passed     99.87%\r\n",
      "  17  CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus         0.00%    passed     96.29%\r\n",
      "  18  CDS_HanXRQr2_Chr02g0065251.1_Helianthus_annuus         0.00%    passed     97.96%\r\n",
      "  19  CDS_HanXRQr2_Chr15g0708181.1_Helianthus_annuus         0.00%    passed     36.15%\r\n",
      "  20  CDS_HanXRQr2_Chr04g0158231.1_Helianthus_annuus         0.00%    passed     97.41%\r\n",
      "  21  CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus         0.00%    passed     96.29%\r\n",
      "  22  Bo8g101500.1_Brassica_oleracea                         0.00%    passed     99.09%\r\n",
      "  23  Bo7g119230.1_Brassica_oleracea                         0.00%    passed     94.44%\r\n",
      "  24  Bo2g088760.1_Brassica_oleracea                        37.10%    passed     30.57%\r\n",
      "  25  Bo1g002490.1_Brassica_oleracea                         0.00%    passed     81.71%\r\n",
      "  26  Bo2g091120.1_Brassica_oleracea                         0.00%    passed     85.33%\r\n",
      "  27  Bo4g147120.1_Brassica_oleracea                         0.00%    passed     97.16%\r\n",
      "  28  Bo6g083750.1_Brassica_oleracea                         0.00%    passed     87.95%\r\n",
      "  29  MD10G1029400_Mdomestica_491_v1                         0.00%    passed     96.32%\r\n",
      "  30  MD15G1340900_Mdomestica_491_v1                         0.00%    passed     99.93%\r\n",
      "  31  MD01G1001000_Mdomestica_491_v1                         0.00%    passed     97.71%\r\n",
      "  32  RHC11H1G0771.2.1.cds.1_Solanum_tuberosum_rh8903916     0.00%    passed     73.74%\r\n",
      "  33  RHC11H1G0775.2.1.cds.1_Solanum_tuberosum_rh8903916     0.00%    passed     99.13%\r\n",
      "  34  RHC04H1G2339.2.1.cds.1_Solanum_tuberosum_rh8903916     0.00%    passed     96.70%\r\n",
      "  35  KJB60013_Gossypium_raimondii                           0.00%    passed     58.49%\r\n",
      "  36  KJB57150_Gossypium_raimondii                           0.00%    passed     96.77%\r\n",
      "  37  KJB71292_Gossypium_raimondii                           0.00%    passed     97.58%\r\n",
      "  38  KZN08747_Daucus_carota                                 0.00%    passed     97.48%\r\n",
      "  39  KZN08752_Daucus_carota                                 0.00%    passed     97.42%\r\n",
      "  40  ERN08734_Amborella_trichopoda                          0.00%    passed     99.93%\r\n",
      "  41  HORVU7Hr1G102010.1_Hordeum_vulgare                     0.00%    passed     81.49%\r\n",
      "  42  HORVU4Hr1G055410.2_Hordeum_vulgare                     0.00%    passed     99.65%\r\n",
      "  43  HORVU4Hr1G055410.3_Hordeum_vulgare                     0.00%    passed     99.55%\r\n",
      "  44  HORVU7Hr1G101980.1_Hordeum_vulgare                     0.00%    passed     99.62%\r\n",
      "  45  HORVU4Hr1G020010.2_Hordeum_vulgare                    35.48%    passed     97.76%\r\n",
      "  46  HORVU4Hr1G020010.1_Hordeum_vulgare                     0.00%    passed     87.42%\r\n",
      "  47  Aco010349.1.mrna1_Ananas_comosus                       0.00%    failed      0.73%\r\n",
      "  48  Aco016990.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.62%\r\n",
      "  49  Aco029478.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.59%\r\n",
      "  50  Aco003193.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.59%\r\n",
      "  51  Aco019928.1.mrna1.cds1_Ananas_comosus                  8.06%    passed     45.52%\r\n",
      "  52  Aco024757.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     95.27%\r\n",
      "  53  Aco024977.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     87.00%\r\n",
      "  54  Aco018685.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     98.42%\r\n",
      "  55  Aco031263.1.mrna1.cds1_Ananas_comosus                 46.77%    passed     94.17%\r\n",
      "  56  PEQU_28350.1_Phalaenopsis_equestris                    0.00%    passed     61.70%\r\n",
      "  57  ONH90092_Prunus_persica                                0.00%    passed     78.62%\r\n",
      "  58  ONH90042_Prunus_persica                                0.00%    passed     99.68%\r\n",
      "  59  ONH90434_Prunus_persica                               30.65%    passed     99.92%\r\n",
      "  60  ONH90087_Prunus_persica                                0.00%    passed     98.76%\r\n",
      "  61  ONH90046_Prunus_persica                                0.00%    passed     99.68%\r\n",
      "  62  ONH89505_Prunus_persica                                0.00%    passed     59.65%\r\n",
      "  63  ONI01825_Prunus_persica                                0.00%    passed     89.38%\r\n",
      "  64  Os11t0168000-01_Oryza_sativa                           0.00%    passed     79.35%\r\n",
      "  65  Os03t0296600-01_Oryza_sativa                           0.00%    passed     95.18%\r\n",
      "  66  EES16714_Sorghum_bicolor                               0.00%    passed     32.43%\r\n",
      "  67  EER95019_Sorghum_bicolor                               0.00%    passed     92.72%\r\n",
      "  68  EOX92363_Theobroma_cacao                               0.00%    passed     97.61%\r\n",
      "  69  EOY04380_Theobroma_cacao                               1.61%    passed     59.62%\r\n",
      "  70  EOY04381_Theobroma_cacao                               1.61%    passed     79.86%\r\n",
      "  71  EOY18917_Theobroma_cacao                               1.61%    passed     41.17%\r\n",
      "  72  EOY16769_Theobroma_cacao                               0.00%    passed     95.23%\r\n",
      "  73  EOY27508_Theobroma_cacao                               0.00%    passed     97.93%\r\n",
      "  74  Ma08_p15310.1_Musa_acuminata                           0.00%    passed     92.61%\r\n",
      "  75  Ma06_p27500.1_Musa_acuminata                           0.00%    passed     84.81%\r\n",
      "  76  Solyc04g071640.1.1.1_Solanum_lycopersicum              0.00%    passed     80.53%\r\n",
      "  77  Solyc11g005500.1.1.1_Solanum_lycopersicum              0.00%    passed     93.00%\r\n",
      "  78  Solyc11g005540.1.1.1_Solanum_lycopersicum              0.00%    passed     89.36%\r\n",
      "  79  itb06g24100.t1_Ipomoea_triloba                         0.00%    passed     99.47%\r\n",
      "  80  itb04g21400.t1_Ipomoea_triloba                         0.00%    passed     84.52%\r\n",
      "  81  itb05g22120.t1_Ipomoea_triloba                         0.00%    passed     98.07%\r\n",
      "  82  OIW19998_Lupinus_angustifolius                         0.00%    passed     98.55%\r\n",
      "  83  OIW19997_Lupinus_angustifolius                        12.90%    passed     67.80%\r\n",
      "  84  OIW20030_Lupinus_angustifolius                         0.00%    passed     86.94%\r\n",
      "  85  Csa16g040920.1.cds1_Camelina_sativa                    0.00%    passed     76.19%\r\n",
      "  86  Csa07g048320.1.cds1_Camelina_sativa                    0.00%    passed     76.19%\r\n",
      "  87  Csa12g002210.1.cds1_Camelina_sativa                    0.00%    passed     47.34%\r\n",
      "  88  Csa07g048380.1.cds1_Camelina_sativa                   87.10%    passed     67.59%\r\n",
      "  89  Csa10g002090.1.cds1_Camelina_sativa                    0.00%    passed     47.13%\r\n",
      "  90  Csa16g043520.1.cds1_Camelina_sativa                    0.00%    passed     98.04%\r\n",
      "  91  Csa07g051890.1.cds1_Camelina_sativa                    0.00%    passed     95.47%\r\n",
      "  92  Csa09g081500.1.cds1_Camelina_sativa                    0.00%    passed     76.19%\r\n",
      "  93  Csa09g086140.1.cds1_Camelina_sativa                    0.00%    passed     98.04%\r\n",
      "  94  Csa11g002350.1.cds1_Camelina_sativa                    0.00%    passed     48.98%\r\n",
      "  95  Csa09g086150.1.cds1_Camelina_sativa                    0.00%    passed     98.28%\r\n",
      "  96  Csa27089s010.1.cds1_Camelina_sativa                   16.13%    passed     70.13%\r\n",
      "  97  Csa16g043530.1_Camelina_sativa                        14.52%    failed      0.94%\r\n",
      "  98  Csa07g051880.1_Camelina_sativa                        24.19%    passed     72.75%\r\n",
      "  99  FCD_00011687-RA_cds_Ficus_carica                       0.00%    passed     99.33%\r\n",
      " 100  FCD_00028685-RA_cds_Ficus_carica                       0.00%    passed     99.56%\r\n",
      " 101  FCD_00030006-RA_cds_Ficus_carica                       0.00%    passed     99.56%\r\n",
      " 102  FCD_00011685-RA_cds_Ficus_carica                       0.00%    passed     93.28%\r\n",
      " 103  RZC48302_Papaver_somniferum                            0.00%    passed     88.68%\r\n",
      " 104  RZC49191_Papaver_somniferum                           16.13%    passed     54.44%\r\n",
      " 105  RZC49225_Papaver_somniferum                            0.00%    passed     78.31%\r\n",
      " 106  RZC55249_Papaver_somniferum                            0.00%    passed     90.15%\r\n",
      " 107  RZC55281_Papaver_somniferum                            0.00%    passed     78.31%\r\n",
      " 108  RZC57082_Papaver_somniferum                            0.00%    passed     76.94%\r\n",
      " 109  RZC57075_Papaver_somniferum                           27.42%    passed     73.17%\r\n",
      " 110  RZC57081_Papaver_somniferum                            0.00%    passed     76.26%\r\n",
      " 111  RZC62100_Papaver_somniferum                            0.00%    passed     56.74%\r\n",
      " 112  RZC62099_Papaver_somniferum                            0.00%    passed     97.17%\r\n",
      " 113  RZC68710_Papaver_somniferum                            0.00%    passed     97.73%\r\n",
      " 114  RZC68605_Papaver_somniferum                           53.23%    passed     98.34%\r\n",
      " 115  RZC68602_Papaver_somniferum                            0.00%    passed     99.11%\r\n",
      " 116  PRQ49244_Rosa_chinensis                                0.00%    passed     75.27%\r\n",
      " 117  PRQ49241_Rosa_chinensis                                0.00%    passed     94.70%\r\n",
      " 118  PRQ25531_Rosa_chinensis                                0.00%    passed     96.26%\r\n",
      " 119  PRQ24668_Rosa_chinensis                                0.00%    passed     99.16%\r\n",
      " 120  PRQ25532_Rosa_chinensis                                0.00%    passed     97.22%\r\n",
      " 121  PRQ17678_Rosa_chinensis                                0.00%    passed     97.20%\r\n",
      " 122  Vradi07g02140.1_Vigna_radiata                          0.00%    passed     99.48%\r\n",
      " 123  Vradi01g10860.1_Vigna_radiata                          0.00%    passed     81.19%\r\n",
      " 124  OIT31467_Nicotiana_attenuata                           0.00%    passed     89.82%\r\n",
      " 125  ESW31795_Phaseolus_vulgaris                            0.00%    passed     83.57%\r\n",
      " 126  ESW18393_Phaseolus_vulgaris                            1.61%    passed     98.97%\r\n",
      " 127  ESW28268_Phaseolus_vulgaris                            0.00%    passed     99.71%\r\n",
      " 128  KEH21230_Medicago_truncatula                           0.00%    passed     97.89%\r\n",
      " 129  AES70181_Medicago_truncatula                           0.00%    passed     99.74%\r\n",
      " 130  AES70191_Medicago_truncatula                          16.13%    passed     88.69%\r\n",
      " 131  OAY36032_Manihot_esculenta                             0.00%    passed     97.10%\r\n",
      " 132  OAY36033_Manihot_esculenta                             0.00%    passed     98.93%\r\n",
      " 133  KRH63270_Glycine_max                                   0.00%    passed     98.59%\r\n",
      " 134  KRH59512_Glycine_max                                   0.00%    passed     93.27%\r\n",
      " 135  KRH54635_Glycine_max                                   0.00%    passed     99.78%\r\n",
      " 136  KRH43377_Glycine_max                                   0.00%    passed     92.08%\r\n",
      " 137  KRH39377_Glycine_max                                   0.00%    passed     93.22%\r\n",
      " 138  KRH35123_Glycine_max                                  22.58%    passed     99.73%\r\n",
      " 139  KRG98863_Glycine_max                                  12.90%    passed     96.11%\r\n",
      " 140  KRG91676_Glycine_max                                   0.00%    passed     99.91%\r\n",
      " 141  KMT04640_Beta_vulgaris                                 0.00%    passed     99.94%\r\n",
      " 142  KMT16083_Beta_vulgaris                                 0.00%    passed     99.75%\r\n",
      " 143  DRNTG_09031.1.cds1_Dioscorea_rotundata                 0.00%    passed     59.94%\r\n",
      " 144  FtPinG0007730800.01.T01_Fagopyrum_tataricum            0.00%    passed     86.27%\r\n",
      " 145  KGN64838_Cucumis_sativus                               0.00%    passed     97.24%\r\n",
      " 146  KGN64837_Cucumis_sativus                               0.00%    passed     91.84%\r\n",
      " 147  KGN46138_Cucumis_sativus                               0.00%    passed     68.45%\r\n",
      " 148  cds.evm.model.01.1369_Cannabis_sativa_female           0.00%    passed     88.49%\r\n",
      " 149  cds.evm.model.04.1284_Cannabis_sativa_female           0.00%    passed     99.96%\r\n",
      " 150  cds.Jr15_01190_p1_Juglans_regia                        0.00%    passed     77.69%\r\n",
      " 151  VIT_09s0002g01570.t01_Vitis_vinifera                   0.00%    passed     75.69%\r\n",
      " 152  VIT_07s0031g02070.t01_Vitis_vinifera                   0.00%    passed     98.09%\r\n",
      " 153  OE9A117531C1_Olea_europaea                             0.00%    passed     95.81%\r\n",
      " 154  NC1G0017660.1_cds_Nymphaea_colorata                    0.00%    passed     75.94%\r\n",
      " 155  NC11G0118510.1_cds_Nymphaea_colorata                   0.00%    passed     73.91%\r\n",
      " 156  NC4G0237690.1_cds_Nymphaea_colorata                   38.71%    passed     62.32%\r\n",
      " 157  Bradi2g28070.1.p_Bdistachyon_314_v3                    0.00%    passed     92.44%\r\n",
      " 158  Bradi4g24640.1.p_Bdistachyon_314_v3                    0.00%    passed     12.10%\r\n",
      " 159  Bradi1g65190.1.p_Bdistachyon_314_v3                    0.00%    passed     97.73%\r\n",
      " 160  ESR47720_Citrus_clementina                             0.00%    passed     99.99%\r\n",
      " 161  Zm00001eb396810_P001_Zea_mays                          0.00%    passed     88.08%\r\n",
      " 162  FvH4_2g18700.t1_Fvesca_677_v4                          0.00%    passed     73.12%\r\n",
      " 163  FvH4_2g18910.t1_Fvesca_677_v4                          1.61%    passed     99.89%\r\n",
      " 164  FvH4_2g13400.t1_Fvesca_677_v4                          0.00%    passed     98.16%\r\n",
      " 165  FvH4_2g13390.t1_Fvesca_677_v4                          0.00%    passed     98.16%\r\n",
      " 166  FvH4_2g18870.t1_Fvesca_677_v4                          0.00%    passed     73.12%\r\n",
      " 167  FvH4_4g25660.t1_Fvesca_677_v4                          0.00%    passed     92.62%\r\n",
      " 168  FvH4_1g23830.t1_Fvesca_677_v4                          0.00%    passed     90.58%\r\n",
      " 169  FvH4_5g02500.t1_Fvesca_677_v4                          0.00%    passed     96.96%\r\n",
      " 170  Kaladp0040s0523.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    0.00%    passed     96.07%\r\n",
      " 171  Kaladp0026s0134.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    0.00%    passed     99.94%\r\n",
      " 172  Kaladp0515s0048.1.v1.1_Kalanchoe_fedtschenkoi         16.13%    passed     16.20%\r\n",
      " 173  Kaladp0033s0187.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    0.00%    passed     98.86%\r\n",
      " 174  ESQ53189_Eutrema_salsugineum                           0.00%    passed     46.99%\r\n",
      " 175  ESQ46086_Eutrema_salsugineum                           0.00%    passed     51.90%\r\n",
      " 176  ESQ46085_Eutrema_salsugineum                           0.00%    passed     86.69%\r\n",
      " 177  ESQ27444_Eutrema_salsugineum                           0.00%    passed     54.96%\r\n",
      " 178  SIN_1014976.t.cds1_Sesamum_indicum                     0.00%    passed     20.17%\r\n",
      " 179  SIN_1019387.t.cds1_Sesamum_indicum                     0.00%    passed     86.67%\r\n",
      " 180  CDS_MD10G0023400.1_Malus_domestica_golden              0.00%    passed     96.32%\r\n",
      " 181  CDS_MD01G0004200.1_Malus_domestica_golden              0.00%    passed     97.71%\r\n",
      " 182  TraesCS4D02G109400.1.cds1_Triticum_aestivum            0.00%    passed     99.80%\r\n",
      " 183  TraesCS7D02G440400.1.cds1_Triticum_aestivum            0.00%    passed     96.90%\r\n",
      " 184  TraesCS7A02G451500.1.cds1_Triticum_aestivum            0.00%    passed     74.44%\r\n",
      " 185  TraesCS7A02G451600.1.cds1_Triticum_aestivum            0.00%    passed     98.90%\r\n",
      " 186  TraesCS7B02G351400.1.cds1_Triticum_aestivum            0.00%    passed     79.23%\r\n",
      " 187  TraesCS7B02G351300.1.cds1_Triticum_aestivum            0.00%    passed     99.79%\r\n",
      " 188  TraesCS4B02G190500.1.cds1_Triticum_aestivum            0.00%    passed     98.35%\r\n",
      " 189  TraesCS7D02G440500.1.cds1_Triticum_aestivum            0.00%    passed     83.94%\r\n",
      " 190  TraesCS4B02G111900.1_Triticum_aestivum                 9.68%    passed     94.43%\r\n",
      " 191  TraesCS4A02G208200.1.cds1_Triticum_aestivum            0.00%    passed     99.46%\r\n",
      " 192  TraesCS4A02G113600.1.cds1_Triticum_aestivum            1.61%    passed     96.78%\r\n",
      " 193  AT4G39340.1_Arabidopsis_thaliana                       0.00%    passed     61.29%\r\n",
      " 194  AT2G21750.1_Arabidopsis_thaliana                       0.00%    passed     98.08%\r\n",
      " 195  AT1G76750.1_Arabidopsis_thaliana                       0.00%    passed     80.02%\r\n",
      " 196  AT2G21740.1_Arabidopsis_thaliana                       0.00%    passed     94.23%\r\n",
      " 197  cds-PLY89992.1_Lactuca_sativa                          0.00%    passed     53.68%\r\n",
      " 198  cds-PLY89774.1_Lactuca_sativa                          0.00%    passed     94.23%\r\n",
      " 199  CDP12837_Coffea_canephora                              0.00%    passed     92.36%\r\n",
      " 200  PSS26677_Actinidia_chinensis                           0.00%    passed     79.28%\r\n",
      " 201  jg50396.t1_Fagopyrum_esculentum                        0.00%    passed     67.62%\r\n",
      "WARNING: 3 sequences contain more than 50% gaps/ambiguity\r\n",
      "****  TOTAL                                                  3.20%  2 sequences failed composition chi2 test (p-value<5%; df=19)\r\n",
      "NOTE: CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus is identical to CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD10G0023400.1_Malus_domestica_golden is identical to MD10G1029400_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD01G0004200.1_Malus_domestica_golden is identical to MD01G1001000_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: Aco003193.1.mrna1.cds1_Ananas_comosus is identical to Aco029478.1.mrna1.cds1_Ananas_comosus but kept for subsequent analysis\r\n",
      "NOTE: ONH90046_Prunus_persica is identical to ONH90042_Prunus_persica but kept for subsequent analysis\r\n",
      "NOTE: Csa07g048320.1.cds1_Camelina_sativa is identical to Csa16g040920.1.cds1_Camelina_sativa but kept for subsequent analysis\r\n",
      "NOTE: Csa09g086140.1.cds1_Camelina_sativa is identical to Csa16g043520.1.cds1_Camelina_sativa but kept for subsequent analysis\r\n",
      "NOTE: FCD_00030006-RA_cds_Ficus_carica is identical to FCD_00028685-RA_cds_Ficus_carica but kept for subsequent analysis\r\n",
      "NOTE: RZC55281_Papaver_somniferum is identical to RZC49225_Papaver_somniferum but kept for subsequent analysis\r\n",
      "NOTE: FvH4_2g18870.t1_Fvesca_677_v4 is identical to FvH4_2g18700.t1_Fvesca_677_v4 but kept for subsequent analysis\r\n",
      "NOTE: FvH4_2g13390.t1_Fvesca_677_v4 is identical to FvH4_2g13400.t1_Fvesca_677_v4 but kept for subsequent analysis\r\n",
      "NOTE: 1 identical sequences (see below) will be ignored for subsequent analysis\r\n",
      "NOTE: Csa09g081500.1.cds1_Camelina_sativa (identical to Csa16g040920.1.cds1_Camelina_sativa) is ignored but added at the end\r\n",
      "Alignment was printed to clustalo_gappyout_trimmed_iqtree_bootstrap.uniqueseq.phy\r\n",
      "\r\n",
      "For your convenience alignment with unique sequences printed to clustalo_gappyout_trimmed_iqtree_bootstrap.uniqueseq.phy\r\n",
      "\r\n",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.037 seconds\n",
      "Perform fast likelihood tree search using LG+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -7668.866\n",
      "Optimal log-likelihood: -7668.865\n",
      "Gamma shape alpha: 1.215\n",
      "Parameters optimization took 1 rounds (0.148 sec)\n",
      "Time for fast ML tree search: 2.485 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 26 MB RAM!\n",
      "ModelFinder will test up to 462 protein models (sample size: 62) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            8007.937     397 16809.875    332821.875   17654.347\n",
      "  2  LG+G4         7668.871     398 16133.742    333737.742   16980.342\n",
      "  3  LG+R2         7740.704     399 16279.408    335479.408   17128.134\n",
      "  4  LG+R3         7688.952     401 16179.904    338583.904   17032.885\n",
      "  5  LG+R4         7665.433     403 16136.866    341760.866   16994.101\n",
      "  6  LG+R5         7663.195     405 16136.390    344996.390   16997.880\n",
      " 13  LG+F+G4       7836.101     417 16506.202    365118.202   17393.217\n",
      " 24  WAG+G4        7661.162     398 16118.324    333722.324   16964.924\n",
      " 35  WAG+F+G4      7768.893     417 16371.785    364983.785   17258.801\n",
      " 46  JTT+G4        7621.639     398 16039.279    333643.279   16885.878\n",
      " 57  JTT+F+G4      7754.540     417 16343.080    364955.080   17230.095\n",
      " 68  JTTDCMut+G4   7622.179     398 16040.358    333644.358   16886.958\n",
      " 79  JTTDCMut+F+G4 7752.008     417 16338.016    364950.016   17225.031\n",
      " 90  DCMut+G4      7715.759     398 16227.518    333831.518   17074.117\n",
      "101  DCMut+F+G4    7798.923     417 16431.845    365043.845   17318.860\n",
      "112  VT+G4         7707.716     398 16211.432    333815.432   17058.032\n",
      "123  VT+F+G4       7856.127     417 16546.255    365158.255   17433.270\n",
      "134  PMB+G4        7801.950     398 16399.901    334003.901   17246.500\n",
      "145  PMB+F+G4      7933.969     417 16701.938    365313.938   17588.953\n",
      "156  Blosum62+G4   7802.643     398 16401.286    334005.286   17247.886\n",
      "167  Blosum62+F+G4 7929.747     417 16693.495    365305.495   17580.510\n",
      "178  Dayhoff+G4    7715.134     398 16226.267    333830.267   17072.867\n",
      "189  Dayhoff+F+G4  7798.394     417 16430.787    365042.787   17317.802\n",
      "200  mtREV+G4      7942.102     398 16680.204    334284.204   17526.804\n",
      "211  mtREV+F+G4    7889.765     417 16613.530    365225.530   17500.545\n",
      "222  mtART+G4      8172.205     398 17140.410    334744.410   17987.010\n",
      "233  mtART+F+G4    8191.336     417 17216.671    365828.671   18103.686\n",
      "244  mtZOA+G4      7915.676     398 16627.351    334231.351   17473.951\n",
      "255  mtZOA+F+G4    8004.353     417 16842.707    365454.707   17729.722\n",
      "266  mtMet+G4      7935.352     398 16666.703    334270.703   17513.303\n",
      "277  mtMet+F+G4    7908.745     417 16651.491    365263.491   17538.506\n",
      "288  mtVer+G4      7923.537     398 16643.074    334247.074   17489.673\n",
      "299  mtVer+F+G4    7971.320     417 16776.640    365388.640   17663.655\n",
      "310  mtInv+G4      8030.292     398 16856.584    334460.584   17703.184\n",
      "321  mtInv+F+G4    7897.275     417 16628.551    365240.551   17515.566\n",
      "332  mtMAM+G4      8029.661     398 16855.322    334459.322   17701.921\n",
      "343  mtMAM+F+G4    8004.912     417 16843.825    365455.825   17730.840\n",
      "354  HIVb+G4       7811.464     398 16418.928    334022.928   17265.527\n",
      "365  HIVb+F+G4     7928.473     417 16690.946    365302.946   17577.961\n",
      "376  HIVw+G4       8081.195     398 16958.390    334562.390   17804.990\n",
      "387  HIVw+F+G4     8013.537     417 16861.074    365473.074   17748.089\n",
      "398  FLU+G4        7749.397     398 16294.795    333898.795   17141.394\n",
      "409  FLU+F+G4      7750.573     417 16335.146    364947.146   17222.161\n",
      "420  rtREV+G4      7783.061     398 16362.122    333966.122   17208.722\n",
      "431  rtREV+F+G4    7966.926     417 16767.852    365379.852   17654.867\n",
      "442  cpREV+G4      7714.019     398 16224.038    333828.038   17070.637\n",
      "453  cpREV+F+G4    7880.127     417 16594.255    365206.255   17481.270\n",
      "Akaike Information Criterion:           JTT+G4\n",
      "Corrected Akaike Information Criterion: LG\n",
      "Bayesian Information Criterion:         JTT+G4\n",
      "Best-fit model: JTT+G4 chosen according to BIC\n",
      "\n",
      "All model information printed to clustalo_gappyout_trimmed_iqtree_bootstrap.model.gz\n",
      "CPU time for ModelFinder: 37.045 seconds (0h:0m:37s)\n",
      "Wall-clock time for ModelFinder: 37.149 seconds (0h:0m:37s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 719585)...\n",
      "\n",
      "NOTE: 10 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -7621.639\n",
      "Optimal log-likelihood: -7621.639\n",
      "Gamma shape alpha: 1.365\n",
      "Parameters optimization took 1 rounds (0.146 sec)\n",
      "Computing ML distances based on estimated model parameters... 3.423 sec\n",
      "WARNING: Some pairwise ML distances are too long (saturated)\n",
      "Computing BIONJ tree...\n",
      "0.017 seconds\n",
      "Log-likelihood of BIONJ tree: -7724.036\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 3.593 second\n",
      "Computing log-likelihood of 98 initial trees ... 6.528 seconds\n",
      "Current best score: -7621.639\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -7617.388\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -7588.805\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 3: -7583.130\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 7: -7580.358\n",
      "Iteration 10 / LogL: -7583.208 / Time: 0h:0m:53s\n",
      "Iteration 20 / LogL: -7610.890 / Time: 0h:1m:29s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -7580.358 / CPU time: 85.336\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 21: -7576.555\n",
      "Iteration 30 / LogL: -7579.863 / Time: 0h:2m:2s (0h:6m:25s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 35: -7572.783\n",
      "BETTER TREE FOUND at iteration 36: -7572.782\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 37: -7571.349\n",
      "Iteration 40 / LogL: -7582.418 / Time: 0h:2m:38s (0h:6m:34s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 48: -7566.555\n",
      "Iteration 50 / LogL: -7575.382 / Time: 0h:3m:16s (0h:6m:32s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 59: -7563.060\n",
      "Iteration 60 / LogL: -7564.800 / Time: 0h:3m:50s (0h:6m:26s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 62: -7560.305\n",
      "Iteration 70 / LogL: -7565.942 / Time: 0h:4m:27s (0h:5m:56s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 74: -7560.112\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 75: -7559.471\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 77: -7558.819\n",
      "Iteration 80 / LogL: -7564.479 / Time: 0h:5m:0s (0h:6m:9s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 86: -7558.656\n",
      "BETTER TREE FOUND at iteration 87: -7558.655\n",
      "Iteration 90 / LogL: -7572.449 / Time: 0h:5m:32s (0h:6m:2s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 97: -7558.641\n",
      "Iteration 100 / LogL: -7559.866 / Time: 0h:6m:4s (0h:5m:56s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.966\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 110 / LogL: -7578.342 / Time: 0h:6m:40s (0h:5m:30s left)\n",
      "Iteration 120 / LogL: -7560.559 / Time: 0h:7m:14s (0h:4m:52s left)\n",
      "Iteration 130 / LogL: -7570.538 / Time: 0h:7m:49s (0h:4m:14s left)\n",
      "Iteration 140 / LogL: -7558.656 / Time: 0h:8m:21s (0h:3m:36s left)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 144: -7558.370\n",
      "Iteration 150 / LogL: -7558.644 / Time: 0h:8m:53s (0h:5m:36s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "BETTER TREE FOUND at iteration 154: -7558.366\n",
      "Iteration 160 / LogL: -7565.402 / Time: 0h:9m:28s (0h:5m:36s left)\n",
      "Iteration 170 / LogL: -7560.351 / Time: 0h:10m:0s (0h:4m:58s left)\n",
      "Iteration 180 / LogL: -7566.379 / Time: 0h:10m:35s (0h:4m:22s left)\n",
      "Iteration 190 / LogL: -7558.528 / Time: 0h:11m:7s (0h:3m:46s left)\n",
      "Iteration 200 / LogL: -7560.311 / Time: 0h:11m:40s (0h:3m:10s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.994\n",
      "Iteration 210 / LogL: -7574.094 / Time: 0h:12m:14s (0h:5m:16s left)\n",
      "Iteration 220 / LogL: -7571.154 / Time: 0h:12m:47s (0h:4m:40s left)\n",
      "Iteration 230 / LogL: -7558.995 / Time: 0h:13m:16s (0h:4m:3s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 237: -7558.309\n",
      "Iteration 240 / LogL: -7574.279 / Time: 0h:13m:47s (0h:5m:35s left)\n",
      "Iteration 250 / LogL: -7558.892 / Time: 0h:14m:18s (0h:4m:59s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "Iteration 260 / LogL: -7569.177 / Time: 0h:14m:51s (0h:4m:25s left)\n",
      "Iteration 270 / LogL: -7599.909 / Time: 0h:15m:21s (0h:3m:49s left)\n",
      "Iteration 280 / LogL: -7563.943 / Time: 0h:15m:50s (0h:3m:14s left)\n",
      "Iteration 290 / LogL: -7558.613 / Time: 0h:16m:22s (0h:2m:39s left)\n",
      "Iteration 300 / LogL: -7574.919 / Time: 0h:16m:54s (0h:2m:5s left)\n",
      "Log-likelihood cutoff on original alignment: -7639.095\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.996\n",
      "Iteration 310 / LogL: -7566.105 / Time: 0h:17m:29s (0h:5m:5s left)\n",
      "Iteration 320 / LogL: -7565.764 / Time: 0h:18m:5s (0h:4m:32s left)\n",
      "Iteration 330 / LogL: -7558.558 / Time: 0h:18m:39s (0h:3m:58s left)\n",
      "TREE SEARCH COMPLETED AFTER 338 ITERATIONS / Time: 0h:19m:5s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -7558.309\n",
      "Optimal log-likelihood: -7558.308\n",
      "Gamma shape alpha: 1.392\n",
      "Parameters optimization took 1 rounds (0.226 sec)\n",
      "BEST SCORE FOUND : -7558.308\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 1000 replicates...\n",
      "Testing tree branches by aBayes parametric test...\n",
      "0.987 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file clustalo_gappyout_trimmed_iqtree_bootstrap.splits.nex\n",
      "Total tree length: 42.860\n",
      "\n",
      "Total number of iterations: 338\n",
      "CPU time used for tree search: 1136.850 sec (0h:18m:56s)\n",
      "Wall-clock time used for tree search: 1141.114 sec (0h:19m:1s)\n",
      "Total CPU time used: 1142.768 sec (0h:19m:2s)\n",
      "Total wall-clock time used: 1147.051 sec (0h:19m:7s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file clustalo_gappyout_trimmed_iqtree_bootstrap.splits.nex...\n",
      "200 taxa and 2052 splits.\n",
      "Consensus tree written to clustalo_gappyout_trimmed_iqtree_bootstrap.contree\n",
      "Reading input trees file clustalo_gappyout_trimmed_iqtree_bootstrap.contree\n",
      "Log-likelihood of consensus tree: -7558.370\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                clustalo_gappyout_trimmed_iqtree_bootstrap.iqtree\n",
      "  Maximum-likelihood tree:       clustalo_gappyout_trimmed_iqtree_bootstrap.treefile\n",
      "  Likelihood distances:          clustalo_gappyout_trimmed_iqtree_bootstrap.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          clustalo_gappyout_trimmed_iqtree_bootstrap.splits.nex\n",
      "  Consensus tree:                clustalo_gappyout_trimmed_iqtree_bootstrap.contree\n",
      "  Screen log file:               clustalo_gappyout_trimmed_iqtree_bootstrap.log\n",
      "\n",
      "Date and Time: Sat May 21 13:22:00 2022\n"
     ]
    }
   ],
   "source": [
    "!iqtree -m MFP -s gappyout_trimmed_alig_clustalo.fa --prefix clustalo_gappyout_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d8ceee7",
   "metadata": {},
   "source": [
    "Coverage:\n",
    "\n",
    "![gappyout_trimmed_alig_clustalo.png](gappyout_trimmed_alig_clustalo.png)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "371a45e7",
   "metadata": {},
   "source": [
    "**Strict**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4408ee7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!trimal -in alig_clustalo.fa -out gappyout_trimmed_alig_clustalo.fa -strict"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "442412c8",
   "metadata": {},
   "source": [
    "Coverage:\n",
    "![strict_trimmed_alig_clustalo.png](strict_trimmed_alig_clustalo.png)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7193c51a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.3 for Linux 64-bit built Dec 20 2020\r\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\r\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\r\n",
      "\r\n",
      "Host:    AnnaToiThink (AVX512, FMA3, 31 GB RAM)\r\n",
      "Command: iqtree -m MFP -s strict_trimmed_alig_clustalo.fa --prefix clustalo_strict_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes\r\n",
      "Seed:    947378 (Using SPRNG - Scalable Parallel Random Number Generator)\r\n",
      "Time:    Sat May 21 13:01:53 2022\r\n",
      "Kernel:  AVX+FMA - 1 threads (8 CPU cores detected)\r\n",
      "\r\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 8 cores!\r\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\r\n",
      "\r\n",
      "Reading alignment file strict_trimmed_alig_clustalo.fa ... Fasta format detected\r\n",
      "Alignment most likely contains protein sequences\r\n",
      "Alignment has 201 sequences with 53 columns, 53 distinct patterns\r\n",
      "48 parsimony-informative, 5 singleton sites, 0 constant sites\r\n",
      "                                                     Gap/Ambiguity  Composition  p-value\r\n",
      "   1  QL10p018382_mrna_CDS_2_Quercus_lobata                  3.77%    passed     94.49%\r\n",
      "   2  QL06p031114_mrna_CDS_2_Quercus_lobata                  3.77%    passed     99.75%\r\n",
      "   3  QL10p018398_mrna_CDS_1_Quercus_lobata                  3.77%    passed     99.89%\r\n",
      "   4  QL11p027894_mrna_CDS_1_Quercus_lobata                  3.77%    passed     99.83%\r\n",
      "   5  QL10p018395_mrna_CDS_1_Quercus_lobata                  3.77%    passed     99.85%\r\n",
      "   6  QL05p056248_mrna_Quercus_lobata                       33.96%    failed      3.65%\r\n",
      "   7  QL10p018379_mrna_CDS_1_Quercus_lobata                  3.77%    passed     99.85%\r\n",
      "   8  QL06p031123_mrna_CDS_1_Quercus_lobata                  3.77%    passed     99.96%\r\n",
      "   9  PNT58448_Populus_trichocarpa                           3.77%    passed     94.66%\r\n",
      "  10  PNT57551_Populus_trichocarpa                           3.77%    passed     81.12%\r\n",
      "  11  PNT57550_Populus_trichocarpa                           3.77%    passed     99.84%\r\n",
      "  12  PNT46426_Populus_trichocarpa                           3.77%    passed     78.22%\r\n",
      "  13  PNT20170_Populus_trichocarpa                           3.77%    passed     68.58%\r\n",
      "  14  PNT12323_Populus_trichocarpa                          73.58%    passed     93.08%\r\n",
      "  15  CDS_HanXRQr2_Chr12g0556091.1_Helianthus_annuus         3.77%    passed     97.05%\r\n",
      "  16  CDS_HanXRQr2_Chr15g0699451.1_Helianthus_annuus         3.77%    passed     98.91%\r\n",
      "  17  CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus         3.77%    passed     93.75%\r\n",
      "  18  CDS_HanXRQr2_Chr02g0065251.1_Helianthus_annuus         3.77%    passed     99.11%\r\n",
      "  19  CDS_HanXRQr2_Chr15g0708181.1_Helianthus_annuus         5.66%    passed     67.01%\r\n",
      "  20  CDS_HanXRQr2_Chr04g0158231.1_Helianthus_annuus         3.77%    passed     99.89%\r\n",
      "  21  CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus         3.77%    passed     93.75%\r\n",
      "  22  Bo8g101500.1_Brassica_oleracea                         3.77%    passed     99.86%\r\n",
      "  23  Bo7g119230.1_Brassica_oleracea                         3.77%    passed     95.14%\r\n",
      "  24  Bo2g088760.1_Brassica_oleracea                        39.62%    passed     54.51%\r\n",
      "  25  Bo1g002490.1_Brassica_oleracea                         3.77%    passed     89.51%\r\n",
      "  26  Bo2g091120.1_Brassica_oleracea                         3.77%    passed     95.93%\r\n",
      "  27  Bo4g147120.1_Brassica_oleracea                         3.77%    passed     96.89%\r\n",
      "  28  Bo6g083750.1_Brassica_oleracea                         3.77%    passed     93.80%\r\n",
      "  29  MD10G1029400_Mdomestica_491_v1                         3.77%    passed     98.78%\r\n",
      "  30  MD15G1340900_Mdomestica_491_v1                         3.77%    passed     99.69%\r\n",
      "  31  MD01G1001000_Mdomestica_491_v1                         3.77%    passed     99.89%\r\n",
      "  32  RHC11H1G0771.2.1.cds.1_Solanum_tuberosum_rh8903916     3.77%    passed     89.96%\r\n",
      "  33  RHC11H1G0775.2.1.cds.1_Solanum_tuberosum_rh8903916     3.77%    passed     98.70%\r\n",
      "  34  RHC04H1G2339.2.1.cds.1_Solanum_tuberosum_rh8903916     3.77%    passed     99.45%\r\n",
      "  35  KJB60013_Gossypium_raimondii                           3.77%    passed     96.67%\r\n",
      "  36  KJB57150_Gossypium_raimondii                           3.77%    passed     98.00%\r\n",
      "  37  KJB71292_Gossypium_raimondii                           3.77%    passed     97.57%\r\n",
      "  38  KZN08747_Daucus_carota                                 3.77%    passed     97.66%\r\n",
      "  39  KZN08752_Daucus_carota                                 3.77%    passed     97.66%\r\n",
      "  40  ERN08734_Amborella_trichopoda                          3.77%    passed     99.99%\r\n",
      "  41  HORVU7Hr1G102010.1_Hordeum_vulgare                     0.00%    passed     85.56%\r\n",
      "  42  HORVU4Hr1G055410.2_Hordeum_vulgare                     0.00%    passed     99.86%\r\n",
      "  43  HORVU4Hr1G055410.3_Hordeum_vulgare                     0.00%    passed     99.86%\r\n",
      "  44  HORVU7Hr1G101980.1_Hordeum_vulgare                     0.00%    passed     99.86%\r\n",
      "  45  HORVU4Hr1G020010.2_Hordeum_vulgare                    43.40%    passed     99.88%\r\n",
      "  46  HORVU4Hr1G020010.1_Hordeum_vulgare                     3.77%    passed     95.86%\r\n",
      "  47  Aco010349.1.mrna1_Ananas_comosus                       0.00%    failed      0.02%\r\n",
      "  48  Aco016990.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.88%\r\n",
      "  49  Aco029478.1.mrna1.cds1_Ananas_comosus                  3.77%    passed     99.98%\r\n",
      "  50  Aco003193.1.mrna1.cds1_Ananas_comosus                  3.77%    passed     99.98%\r\n",
      "  51  Aco019928.1.mrna1.cds1_Ananas_comosus                 15.09%    passed      8.25%\r\n",
      "  52  Aco024757.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.56%\r\n",
      "  53  Aco024977.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     92.23%\r\n",
      "  54  Aco018685.1.mrna1.cds1_Ananas_comosus                  0.00%    passed     99.61%\r\n",
      "  55  Aco031263.1.mrna1.cds1_Ananas_comosus                 47.17%    passed     96.68%\r\n",
      "  56  PEQU_28350.1_Phalaenopsis_equestris                    0.00%    passed     93.66%\r\n",
      "  57  ONH90092_Prunus_persica                                3.77%    passed     95.40%\r\n",
      "  58  ONH90042_Prunus_persica                                3.77%    passed     99.80%\r\n",
      "  59  ONH90434_Prunus_persica                               35.85%    passed     97.77%\r\n",
      "  60  ONH90087_Prunus_persica                                3.77%    passed     99.15%\r\n",
      "  61  ONH90046_Prunus_persica                                3.77%    passed     99.80%\r\n",
      "  62  ONH89505_Prunus_persica                                3.77%    passed     98.20%\r\n",
      "  63  ONI01825_Prunus_persica                                3.77%    passed     99.51%\r\n",
      "  64  Os11t0168000-01_Oryza_sativa                           3.77%    passed     97.25%\r\n",
      "  65  Os03t0296600-01_Oryza_sativa                           0.00%    passed     99.65%\r\n",
      "  66  EES16714_Sorghum_bicolor                               3.77%    passed     70.05%\r\n",
      "  67  EER95019_Sorghum_bicolor                               0.00%    passed     84.24%\r\n",
      "  68  EOX92363_Theobroma_cacao                               3.77%    passed     99.36%\r\n",
      "  69  EOY04380_Theobroma_cacao                               5.66%    passed     87.78%\r\n",
      "  70  EOY04381_Theobroma_cacao                               5.66%    passed     52.54%\r\n",
      "  71  EOY18917_Theobroma_cacao                               3.77%    passed     28.97%\r\n",
      "  72  EOY16769_Theobroma_cacao                               3.77%    passed     98.96%\r\n",
      "  73  EOY27508_Theobroma_cacao                               3.77%    passed     99.59%\r\n",
      "  74  Ma08_p15310.1_Musa_acuminata                           0.00%    passed     99.92%\r\n",
      "  75  Ma06_p27500.1_Musa_acuminata                           3.77%    passed     99.41%\r\n",
      "  76  Solyc04g071640.1.1.1_Solanum_lycopersicum              3.77%    passed     88.60%\r\n",
      "  77  Solyc11g005500.1.1.1_Solanum_lycopersicum              3.77%    passed     92.94%\r\n",
      "  78  Solyc11g005540.1.1.1_Solanum_lycopersicum              3.77%    passed     97.55%\r\n",
      "  79  itb06g24100.t1_Ipomoea_triloba                         3.77%    passed     99.60%\r\n",
      "  80  itb04g21400.t1_Ipomoea_triloba                         3.77%    passed     97.69%\r\n",
      "  81  itb05g22120.t1_Ipomoea_triloba                         1.89%    passed     99.11%\r\n",
      "  82  OIW19998_Lupinus_angustifolius                         3.77%    passed     99.99%\r\n",
      "  83  OIW19997_Lupinus_angustifolius                        15.09%    passed     95.34%\r\n",
      "  84  OIW20030_Lupinus_angustifolius                         3.77%    passed     89.26%\r\n",
      "  85  Csa16g040920.1.cds1_Camelina_sativa                    3.77%    passed     93.80%\r\n",
      "  86  Csa07g048320.1.cds1_Camelina_sativa                    3.77%    passed     93.80%\r\n",
      "  87  Csa12g002210.1.cds1_Camelina_sativa                    3.77%    passed     95.13%\r\n",
      "  88  Csa07g048380.1.cds1_Camelina_sativa                   86.79%    passed     66.06%\r\n",
      "  89  Csa10g002090.1.cds1_Camelina_sativa                    3.77%    passed     95.14%\r\n",
      "  90  Csa16g043520.1.cds1_Camelina_sativa                    3.77%    passed     99.48%\r\n",
      "  91  Csa07g051890.1.cds1_Camelina_sativa                    3.77%    passed     99.21%\r\n",
      "  92  Csa09g081500.1.cds1_Camelina_sativa                    3.77%    passed     93.80%\r\n",
      "  93  Csa09g086140.1.cds1_Camelina_sativa                    3.77%    passed     99.48%\r\n",
      "  94  Csa11g002350.1.cds1_Camelina_sativa                    3.77%    passed     96.01%\r\n",
      "  95  Csa09g086150.1.cds1_Camelina_sativa                    3.77%    passed     99.67%\r\n",
      "  96  Csa27089s010.1.cds1_Camelina_sativa                   15.09%    passed     77.23%\r\n",
      "  97  Csa16g043530.1_Camelina_sativa                        16.98%    failed      0.17%\r\n",
      "  98  Csa07g051880.1_Camelina_sativa                        28.30%    passed     85.38%\r\n",
      "  99  FCD_00011687-RA_cds_Ficus_carica                       3.77%    passed     99.93%\r\n",
      " 100  FCD_00028685-RA_cds_Ficus_carica                       3.77%    passed     99.81%\r\n",
      " 101  FCD_00030006-RA_cds_Ficus_carica                       3.77%    passed     99.81%\r\n",
      " 102  FCD_00011685-RA_cds_Ficus_carica                       3.77%    passed     86.40%\r\n",
      " 103  RZC48302_Papaver_somniferum                            3.77%    passed     98.70%\r\n",
      " 104  RZC49191_Papaver_somniferum                           22.64%    passed     77.61%\r\n",
      " 105  RZC49225_Papaver_somniferum                            3.77%    passed     98.70%\r\n",
      " 106  RZC55249_Papaver_somniferum                            3.77%    passed     99.69%\r\n",
      " 107  RZC55281_Papaver_somniferum                            3.77%    passed     98.70%\r\n",
      " 108  RZC57082_Papaver_somniferum                            3.77%    passed     98.71%\r\n",
      " 109  RZC57075_Papaver_somniferum                           26.42%    passed     92.00%\r\n",
      " 110  RZC57081_Papaver_somniferum                            3.77%    passed     98.89%\r\n",
      " 111  RZC62100_Papaver_somniferum                            3.77%    passed     59.36%\r\n",
      " 112  RZC62099_Papaver_somniferum                            3.77%    passed     91.22%\r\n",
      " 113  RZC68710_Papaver_somniferum                            3.77%    passed     89.75%\r\n",
      " 114  RZC68605_Papaver_somniferum                           58.49%    passed     87.83%\r\n",
      " 115  RZC68602_Papaver_somniferum                            3.77%    passed     99.90%\r\n",
      " 116  PRQ49244_Rosa_chinensis                                3.77%    passed     81.08%\r\n",
      " 117  PRQ49241_Rosa_chinensis                                3.77%    passed     97.25%\r\n",
      " 118  PRQ25531_Rosa_chinensis                                3.77%    passed     95.78%\r\n",
      " 119  PRQ24668_Rosa_chinensis                                3.77%    passed     99.17%\r\n",
      " 120  PRQ25532_Rosa_chinensis                                3.77%    passed     99.09%\r\n",
      " 121  PRQ17678_Rosa_chinensis                                3.77%    passed     93.00%\r\n",
      " 122  Vradi07g02140.1_Vigna_radiata                          3.77%    passed     98.93%\r\n",
      " 123  Vradi01g10860.1_Vigna_radiata                          3.77%    passed     55.38%\r\n",
      " 124  OIT31467_Nicotiana_attenuata                           3.77%    passed     97.85%\r\n",
      " 125  ESW31795_Phaseolus_vulgaris                            3.77%    passed     82.85%\r\n",
      " 126  ESW18393_Phaseolus_vulgaris                            3.77%    passed     92.20%\r\n",
      " 127  ESW28268_Phaseolus_vulgaris                            3.77%    passed     98.66%\r\n",
      " 128  KEH21230_Medicago_truncatula                           3.77%    passed     89.67%\r\n",
      " 129  AES70181_Medicago_truncatula                           3.77%    passed     91.63%\r\n",
      " 130  AES70191_Medicago_truncatula                          16.98%    passed     84.51%\r\n",
      " 131  OAY36032_Manihot_esculenta                             3.77%    passed     99.82%\r\n",
      " 132  OAY36033_Manihot_esculenta                             3.77%    passed     96.19%\r\n",
      " 133  KRH63270_Glycine_max                                   3.77%    passed     97.54%\r\n",
      " 134  KRH59512_Glycine_max                                   3.77%    passed     55.77%\r\n",
      " 135  KRH54635_Glycine_max                                   3.77%    passed     98.76%\r\n",
      " 136  KRH43377_Glycine_max                                   3.77%    passed     56.79%\r\n",
      " 137  KRH39377_Glycine_max                                   3.77%    passed     94.77%\r\n",
      " 138  KRH35123_Glycine_max                                  20.75%    passed     99.61%\r\n",
      " 139  KRG98863_Glycine_max                                  13.21%    passed     92.76%\r\n",
      " 140  KRG91676_Glycine_max                                   3.77%    passed     99.88%\r\n",
      " 141  KMT04640_Beta_vulgaris                                 3.77%    passed     99.41%\r\n",
      " 142  KMT16083_Beta_vulgaris                                 3.77%    passed     99.85%\r\n",
      " 143  DRNTG_09031.1.cds1_Dioscorea_rotundata                 0.00%    passed     80.59%\r\n",
      " 144  FtPinG0007730800.01.T01_Fagopyrum_tataricum            3.77%    passed     85.31%\r\n",
      " 145  KGN64838_Cucumis_sativus                               3.77%    passed     95.92%\r\n",
      " 146  KGN64837_Cucumis_sativus                               3.77%    passed     74.65%\r\n",
      " 147  KGN46138_Cucumis_sativus                               3.77%    passed     86.52%\r\n",
      " 148  cds.evm.model.01.1369_Cannabis_sativa_female           3.77%    passed     97.25%\r\n",
      " 149  cds.evm.model.04.1284_Cannabis_sativa_female           3.77%    passed     98.94%\r\n",
      " 150  cds.Jr15_01190_p1_Juglans_regia                        5.66%    passed     89.86%\r\n",
      " 151  VIT_09s0002g01570.t01_Vitis_vinifera                   3.77%    passed     81.86%\r\n",
      " 152  VIT_07s0031g02070.t01_Vitis_vinifera                   3.77%    passed     99.94%\r\n",
      " 153  OE9A117531C1_Olea_europaea                             3.77%    passed     98.63%\r\n",
      " 154  NC1G0017660.1_cds_Nymphaea_colorata                    3.77%    passed     95.63%\r\n",
      " 155  NC11G0118510.1_cds_Nymphaea_colorata                   3.77%    passed     88.57%\r\n",
      " 156  NC4G0237690.1_cds_Nymphaea_colorata                   41.51%    passed     74.91%\r\n",
      " 157  Bradi2g28070.1.p_Bdistachyon_314_v3                    0.00%    passed     97.46%\r\n",
      " 158  Bradi4g24640.1.p_Bdistachyon_314_v3                    3.77%    passed     56.43%\r\n",
      " 159  Bradi1g65190.1.p_Bdistachyon_314_v3                    0.00%    passed     98.72%\r\n",
      " 160  ESR47720_Citrus_clementina                             3.77%    passed     98.90%\r\n",
      " 161  Zm00001eb396810_P001_Zea_mays                          0.00%    passed     81.01%\r\n",
      " 162  FvH4_2g18700.t1_Fvesca_677_v4                          3.77%    passed     83.98%\r\n",
      " 163  FvH4_2g18910.t1_Fvesca_677_v4                          3.77%    passed     98.82%\r\n",
      " 164  FvH4_2g13400.t1_Fvesca_677_v4                          3.77%    passed     98.33%\r\n",
      " 165  FvH4_2g13390.t1_Fvesca_677_v4                          3.77%    passed     98.33%\r\n",
      " 166  FvH4_2g18870.t1_Fvesca_677_v4                          3.77%    passed     83.98%\r\n",
      " 167  FvH4_4g25660.t1_Fvesca_677_v4                          3.77%    passed     99.67%\r\n",
      " 168  FvH4_1g23830.t1_Fvesca_677_v4                          3.77%    passed     98.21%\r\n",
      " 169  FvH4_5g02500.t1_Fvesca_677_v4                          3.77%    passed     98.67%\r\n",
      " 170  Kaladp0040s0523.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    3.77%    passed     98.33%\r\n",
      " 171  Kaladp0026s0134.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    3.77%    passed     99.42%\r\n",
      " 172  Kaladp0515s0048.1.v1.1_Kalanchoe_fedtschenkoi         18.87%    failed      0.22%\r\n",
      " 173  Kaladp0033s0187.1.v1.1.CDS.1_Kalanchoe_fedtschenkoi    3.77%    passed     98.97%\r\n",
      " 174  ESQ53189_Eutrema_salsugineum                           3.77%    passed     45.56%\r\n",
      " 175  ESQ46086_Eutrema_salsugineum                           3.77%    passed     88.44%\r\n",
      " 176  ESQ46085_Eutrema_salsugineum                           3.77%    passed     99.70%\r\n",
      " 177  ESQ27444_Eutrema_salsugineum                           3.77%    passed     66.76%\r\n",
      " 178  SIN_1014976.t.cds1_Sesamum_indicum                     5.66%    passed     15.22%\r\n",
      " 179  SIN_1019387.t.cds1_Sesamum_indicum                     3.77%    passed     99.15%\r\n",
      " 180  CDS_MD10G0023400.1_Malus_domestica_golden              3.77%    passed     98.78%\r\n",
      " 181  CDS_MD01G0004200.1_Malus_domestica_golden              3.77%    passed     99.89%\r\n",
      " 182  TraesCS4D02G109400.1.cds1_Triticum_aestivum            3.77%    passed     99.95%\r\n",
      " 183  TraesCS7D02G440400.1.cds1_Triticum_aestivum            0.00%    passed     99.17%\r\n",
      " 184  TraesCS7A02G451500.1.cds1_Triticum_aestivum            0.00%    passed     97.80%\r\n",
      " 185  TraesCS7A02G451600.1.cds1_Triticum_aestivum            0.00%    passed     96.55%\r\n",
      " 186  TraesCS7B02G351400.1.cds1_Triticum_aestivum            0.00%    passed     93.23%\r\n",
      " 187  TraesCS7B02G351300.1.cds1_Triticum_aestivum            0.00%    passed     99.78%\r\n",
      " 188  TraesCS4B02G190500.1.cds1_Triticum_aestivum            0.00%    passed     99.86%\r\n",
      " 189  TraesCS7D02G440500.1.cds1_Triticum_aestivum            0.00%    passed     93.23%\r\n",
      " 190  TraesCS4B02G111900.1_Triticum_aestivum                 9.43%    passed     96.71%\r\n",
      " 191  TraesCS4A02G208200.1.cds1_Triticum_aestivum            3.77%    passed     99.92%\r\n",
      " 192  TraesCS4A02G113600.1.cds1_Triticum_aestivum            0.00%    passed     99.62%\r\n",
      " 193  AT4G39340.1_Arabidopsis_thaliana                       3.77%    passed     95.37%\r\n",
      " 194  AT2G21750.1_Arabidopsis_thaliana                       3.77%    passed     99.42%\r\n",
      " 195  AT1G76750.1_Arabidopsis_thaliana                       3.77%    passed     93.80%\r\n",
      " 196  AT2G21740.1_Arabidopsis_thaliana                       3.77%    passed     98.72%\r\n",
      " 197  cds-PLY89992.1_Lactuca_sativa                          3.77%    passed     97.85%\r\n",
      " 198  cds-PLY89774.1_Lactuca_sativa                          3.77%    passed     99.54%\r\n",
      " 199  CDP12837_Coffea_canephora                              3.77%    passed     98.98%\r\n",
      " 200  PSS26677_Actinidia_chinensis                           3.77%    passed     98.66%\r\n",
      " 201  jg50396.t1_Fagopyrum_esculentum                        3.77%    passed     56.88%\r\n",
      "WARNING: 3 sequences contain more than 50% gaps/ambiguity\r\n",
      "****  TOTAL                                                  6.33%  4 sequences failed composition chi2 test (p-value<5%; df=19)\r\n",
      "NOTE: QL10p018379_mrna_CDS_1_Quercus_lobata is identical to QL10p018395_mrna_CDS_1_Quercus_lobata but kept for subsequent analysis\r\n",
      "NOTE: CDS_HanXRQr2_Chr02g0065281.1_Helianthus_annuus is identical to CDS_HanXRQr2_Chr02g0065241.1_Helianthus_annuus but kept for subsequent analysis\r\n",
      "NOTE: Csa10g002090.1.cds1_Camelina_sativa is identical to Bo7g119230.1_Brassica_oleracea but kept for subsequent analysis\r\n",
      "NOTE: Csa16g040920.1.cds1_Camelina_sativa is identical to Bo6g083750.1_Brassica_oleracea but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD10G0023400.1_Malus_domestica_golden is identical to MD10G1029400_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: CDS_MD01G0004200.1_Malus_domestica_golden is identical to MD01G1001000_Mdomestica_491_v1 but kept for subsequent analysis\r\n",
      "NOTE: KZN08752_Daucus_carota is identical to KZN08747_Daucus_carota but kept for subsequent analysis\r\n",
      "NOTE: HORVU4Hr1G055410.3_Hordeum_vulgare is identical to HORVU4Hr1G055410.2_Hordeum_vulgare but kept for subsequent analysis\r\n",
      "NOTE: Aco003193.1.mrna1.cds1_Ananas_comosus is identical to Aco029478.1.mrna1.cds1_Ananas_comosus but kept for subsequent analysis\r\n",
      "NOTE: ONH90046_Prunus_persica is identical to ONH90042_Prunus_persica but kept for subsequent analysis\r\n",
      "NOTE: Csa09g086140.1.cds1_Camelina_sativa is identical to Csa16g043520.1.cds1_Camelina_sativa but kept for subsequent analysis\r\n",
      "NOTE: FCD_00030006-RA_cds_Ficus_carica is identical to FCD_00028685-RA_cds_Ficus_carica but kept for subsequent analysis\r\n",
      "NOTE: RZC49225_Papaver_somniferum is identical to RZC48302_Papaver_somniferum but kept for subsequent analysis\r\n",
      "NOTE: FvH4_2g18870.t1_Fvesca_677_v4 is identical to FvH4_2g18700.t1_Fvesca_677_v4 but kept for subsequent analysis\r\n",
      "NOTE: FvH4_2g13390.t1_Fvesca_677_v4 is identical to FvH4_2g13400.t1_Fvesca_677_v4 but kept for subsequent analysis\r\n",
      "NOTE: TraesCS7D02G440500.1.cds1_Triticum_aestivum is identical to TraesCS7B02G351400.1.cds1_Triticum_aestivum but kept for subsequent analysis\r\n",
      "NOTE: 5 identical sequences (see below) will be ignored for subsequent analysis\r\n",
      "NOTE: Csa07g048320.1.cds1_Camelina_sativa (identical to Bo6g083750.1_Brassica_oleracea) is ignored but added at the end\r\n",
      "NOTE: Csa09g081500.1.cds1_Camelina_sativa (identical to Bo6g083750.1_Brassica_oleracea) is ignored but added at the end\r\n",
      "NOTE: AT1G76750.1_Arabidopsis_thaliana (identical to Bo6g083750.1_Brassica_oleracea) is ignored but added at the end\r\n",
      "NOTE: TraesCS4B02G190500.1.cds1_Triticum_aestivum (identical to HORVU4Hr1G055410.2_Hordeum_vulgare) is ignored but added at the end\r\n",
      "NOTE: RZC55281_Papaver_somniferum (identical to RZC48302_Papaver_somniferum) is ignored but added at the end\r\n",
      "Alignment was printed to clustalo_strict_trimmed_iqtree_bootstrap.uniqueseq.phy\r\n",
      "\r\n",
      "For your convenience alignment with unique sequences printed to clustalo_strict_trimmed_iqtree_bootstrap.uniqueseq.phy\r\n",
      "\r\n",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.050 seconds\n",
      "Perform fast likelihood tree search using LG+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -5394.722\n",
      "Optimal log-likelihood: -5394.643\n",
      "Gamma shape alpha: 1.112\n",
      "Parameters optimization took 1 rounds (0.247 sec)\n",
      "Time for fast ML tree search: 2.586 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 23 MB RAM!\n",
      "ModelFinder will test up to 462 protein models (sample size: 53) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            5634.104     389 12046.209    315466.209   12812.652\n",
      "  2  LG+G4         5394.809     390 11569.619    316549.619   12338.033\n",
      "  3  LG+R2         5456.270     391 11694.540    318238.540   12464.924\n",
      "  4  LG+R3         5403.439     393 11592.879    321276.879   12367.203\n",
      "  5  LG+R4         5389.721     395 11569.441    324409.441   12347.707\n",
      "  6  LG+R5         5387.894     397 11569.789    327581.789   12351.995\n",
      " 13  LG+F+G4       5576.956     409 11971.912    347351.912   12777.761\n",
      " 16  LG+F+R4       5572.015     414 11972.031    355592.031   12787.732\n",
      " 24  WAG+G4        5405.216     390 11590.432    316570.432   12358.846\n",
      " 27  WAG+R4        5400.421     395 11590.841    324430.841   12369.106\n",
      " 35  WAG+F+G4      5538.253     409 11894.505    347274.505   12700.355\n",
      " 38  WAG+F+R4      5532.784     414 11893.568    355513.568   12709.269\n",
      " 46  JTT+G4        5377.109     390 11534.219    316514.219   12302.633\n",
      " 49  JTT+R4        5371.518     395 11533.036    324373.036   12311.302\n",
      " 57  JTT+F+G4      5525.175     409 11868.351    347248.351   12674.200\n",
      " 60  JTT+F+R4      5518.581     414 11865.162    355485.162   12680.863\n",
      " 68  JTTDCMut+G4   5376.694     390 11533.388    316513.388   12301.802\n",
      " 71  JTTDCMut+R4   5370.957     395 11531.913    324371.913   12310.178\n",
      " 79  JTTDCMut+F+G4 5522.435     409 11862.870    347242.870   12668.719\n",
      " 82  JTTDCMut+F+R4 5515.771     414 11859.542    355479.542   12675.242\n",
      " 90  DCMut+G4      5460.554     390 11701.107    316681.107   12469.521\n",
      " 93  DCMut+R4      5450.508     395 11691.015    324531.015   12469.281\n",
      "101  DCMut+F+G4    5568.994     409 11955.988    347335.988   12761.837\n",
      "104  DCMut+F+R4    5564.956     414 11957.912    355577.912   12773.613\n",
      "112  VT+G4         5437.391     390 11654.782    316634.782   12423.195\n",
      "115  VT+R4         5432.715     395 11655.430    324495.430   12433.696\n",
      "123  VT+F+G4       5600.683     409 12019.365    347399.365   12825.215\n",
      "126  VT+F+R4       5596.170     414 12020.339    355640.339   12836.040\n",
      "134  PMB+G4        5489.061     390 11758.123    316738.123   12526.537\n",
      "137  PMB+R4        5483.874     395 11757.748    324597.748   12536.014\n",
      "145  PMB+F+G4      5649.330     409 12116.660    347496.660   12922.510\n",
      "148  PMB+F+R4      5644.699     414 12117.399    355737.399   12933.100\n",
      "156  Blosum62+G4   5487.788     390 11755.577    316735.577   12523.991\n",
      "159  Blosum62+R4   5482.191     395 11754.383    324594.383   12532.648\n",
      "167  Blosum62+F+G4 5641.787     409 12101.573    347481.573   12907.423\n",
      "170  Blosum62+F+R4 5637.004     414 12102.007    355722.007   12917.708\n",
      "178  Dayhoff+G4    5459.799     390 11699.598    316679.598   12468.011\n",
      "181  Dayhoff+R4    5449.725     395 11689.450    324529.450   12467.715\n",
      "189  Dayhoff+F+G4  5569.309     409 11956.619    347336.619   12762.468\n",
      "192  Dayhoff+F+R4  5564.014     414 11956.028    355576.028   12771.729\n",
      "200  mtREV+G4      5552.747     390 11885.494    316865.494   12653.908\n",
      "203  mtREV+R4      5548.364     395 11886.727    324726.727   12664.992\n",
      "211  mtREV+F+G4    5587.394     409 11992.787    347372.787   12798.636\n",
      "214  mtREV+F+R4    5582.792     414 11993.585    355613.585   12809.286\n",
      "222  mtART+G4      5722.905     390 12225.811    317205.811   12994.225\n",
      "225  mtART+R4      5720.761     395 12231.522    325071.522   13009.787\n",
      "233  mtART+F+G4    5827.479     409 12472.957    347852.957   13278.807\n",
      "236  mtART+F+R4    5823.412     414 12474.823    356094.823   13290.524\n",
      "244  mtZOA+G4      5544.697     390 11869.394    316849.394   12637.808\n",
      "247  mtZOA+R4      5540.936     395 11871.871    324711.871   12650.137\n",
      "255  mtZOA+F+G4    5693.977     409 12205.954    347585.954   13011.803\n",
      "258  mtZOA+F+R4    5687.455     414 12202.910    355822.910   13018.611\n",
      "266  mtMet+G4      5548.837     390 11877.673    316857.673   12646.087\n",
      "269  mtMet+R4      5542.849     395 11875.698    324715.698   12653.963\n",
      "277  mtMet+F+G4    5619.643     409 12057.286    347437.286   12863.135\n",
      "280  mtMet+F+R4    5615.363     414 12058.726    355678.726   12874.427\n",
      "288  mtVer+G4      5542.055     390 11864.109    316844.109   12632.523\n",
      "291  mtVer+R4      5537.802     395 11865.604    324705.604   12643.870\n",
      "299  mtVer+F+G4    5657.177     409 12132.355    347512.355   12938.204\n",
      "302  mtVer+F+R4    5647.864     414 12123.728    355743.728   12939.429\n",
      "310  mtInv+G4      5616.023     390 12012.046    316992.046   12780.460\n",
      "313  mtInv+R4      5612.031     395 12014.062    324854.062   12792.328\n",
      "321  mtInv+F+G4    5628.057     409 12074.113    347454.113   12879.963\n",
      "324  mtInv+F+R4    5623.218     414 12074.436    355694.436   12890.137\n",
      "332  mtMAM+G4      5578.891     390 11937.783    316917.783   12706.197\n",
      "335  mtMAM+R4      5576.840     395 11943.681    324783.681   12721.946\n",
      "343  mtMAM+F+G4    5631.772     409 12081.544    347461.544   12887.393\n",
      "346  mtMAM+F+R4    5629.032     414 12086.064    355706.064   12901.764\n",
      "354  HIVb+G4       5487.060     390 11754.121    316734.121   12522.534\n",
      "357  HIVb+R4       5481.149     395 11752.299    324592.299   12530.564\n",
      "365  HIVb+F+G4     5643.408     409 12104.816    347484.816   12910.666\n",
      "368  HIVb+F+R4     5638.015     414 12104.030    355724.030   12919.731\n",
      "376  HIVw+G4       5677.545     390 12135.090    317115.090   12903.503\n",
      "379  HIVw+R4       5662.150     395 12114.301    324954.301   12892.566\n",
      "387  HIVw+F+G4     5697.074     409 12212.147    347592.147   13017.997\n",
      "390  HIVw+F+R4     5685.183     414 12198.366    355818.366   13014.067\n",
      "398  FLU+G4        5445.913     390 11671.826    316651.826   12440.240\n",
      "401  FLU+R4        5436.698     395 11663.396    324503.396   12441.662\n",
      "409  FLU+F+G4      5509.410     409 11836.819    347216.819   12642.669\n",
      "412  FLU+F+R4      5502.344     414 11832.689    355452.689   12648.389\n",
      "420  rtREV+G4      5474.629     390 11729.258    316709.258   12497.672\n",
      "423  rtREV+R4      5469.686     395 11729.372    324569.372   12507.637\n",
      "431  rtREV+F+G4    5661.708     409 12141.416    347521.416   12947.266\n",
      "434  rtREV+F+R4    5657.379     414 12142.759    355762.759   12958.460\n",
      "442  cpREV+G4      5427.575     390 11635.150    316615.150   12403.564\n",
      "445  cpREV+R4      5421.770     395 11633.539    324473.539   12411.804\n",
      "453  cpREV+F+G4    5613.337     409 12044.673    347424.673   12850.523\n",
      "456  cpREV+F+R4    5607.227     414 12042.455    355662.455   12858.156\n",
      "Akaike Information Criterion:           JTTDCMut+R4\n",
      "Corrected Akaike Information Criterion: LG\n",
      "Bayesian Information Criterion:         JTTDCMut+G4\n",
      "Best-fit model: JTTDCMut+G4 chosen according to BIC\n",
      "\n",
      "All model information printed to clustalo_strict_trimmed_iqtree_bootstrap.model.gz\n",
      "CPU time for ModelFinder: 112.458 seconds (0h:1m:52s)\n",
      "Wall-clock time for ModelFinder: 112.717 seconds (0h:1m:52s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 947378)...\n",
      "\n",
      "NOTE: 9 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -5376.694\n",
      "Optimal log-likelihood: -5376.690\n",
      "Gamma shape alpha: 1.240\n",
      "Parameters optimization took 1 rounds (0.242 sec)\n",
      "Computing ML distances based on estimated model parameters... 5.210 sec\n",
      "WARNING: Some pairwise ML distances are too long (saturated)\n",
      "Computing BIONJ tree...\n",
      "0.025 seconds\n",
      "Log-likelihood of BIONJ tree: -5480.428\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 6.184 second\n",
      "Computing log-likelihood of 98 initial trees ... 7.339 seconds\n",
      "Current best score: -5376.690\n",
      "\n",
      "Do NNI search on 20 best initial trees\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -5370.708\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -5358.683\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 4: -5338.473\n",
      "Iteration 10 / LogL: -5383.184 / Time: 0h:0m:56s\n",
      "Iteration 20 / LogL: -5371.414 / Time: 0h:1m:39s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -5338.473 / CPU time: 93.261\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 22: -5333.088\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 23: -5329.150\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 27: -5328.862\n",
      "Iteration 30 / LogL: -5333.696 / Time: 0h:2m:18s (0h:7m:43s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 37: -5328.742\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 40: -5323.422\n",
      "Iteration 40 / LogL: -5323.422 / Time: 0h:2m:58s (0h:7m:36s left)\n",
      "Iteration 50 / LogL: -5338.135 / Time: 0h:3m:42s (0h:6m:47s left)\n",
      "Log-likelihood cutoff on original alignment: -5408.812\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 57: -5322.780\n",
      "Iteration 60 / LogL: -5325.831 / Time: 0h:4m:28s (0h:7m:21s left)\n",
      "Iteration 70 / LogL: -5339.896 / Time: 0h:5m:4s (0h:6m:24s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 73: -5321.206\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 78: -5320.979\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 79: -5320.826\n",
      "Iteration 80 / LogL: -5325.944 / Time: 0h:5m:43s (0h:7m:10s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 86: -5315.125\n",
      "Iteration 90 / LogL: -5361.065 / Time: 0h:6m:21s (0h:6m:51s left)\n",
      "Iteration 100 / LogL: -5346.927 / Time: 0h:6m:59s (0h:6m:4s left)\n",
      "Log-likelihood cutoff on original alignment: -5408.812\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.970\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 110 / LogL: -5321.154 / Time: 0h:7m:36s (0h:6m:16s left)\n",
      "Iteration 120 / LogL: -5334.245 / Time: 0h:8m:13s (0h:5m:31s left)\n",
      "Iteration 130 / LogL: -5326.261 / Time: 0h:8m:51s (0h:4m:48s left)\n",
      "Iteration 140 / LogL: -5334.888 / Time: 0h:9m:33s (0h:4m:7s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 147: -5314.067\n",
      "Iteration 150 / LogL: -5316.302 / Time: 0h:10m:12s (0h:6m:38s left)\n",
      "Log-likelihood cutoff on original alignment: -5408.812\n",
      "Iteration 160 / LogL: -5334.294 / Time: 0h:10m:53s (0h:5m:57s left)\n",
      "Iteration 170 / LogL: -5328.345 / Time: 0h:11m:31s (0h:5m:14s left)\n",
      "Iteration 180 / LogL: -5328.553 / Time: 0h:12m:10s (0h:4m:33s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 181: -5313.566\n",
      "Iteration 190 / LogL: -5314.701 / Time: 0h:12m:45s (0h:6m:8s left)\n",
      "Iteration 200 / LogL: -5317.697 / Time: 0h:13m:22s (0h:5m:26s left)\n",
      "Log-likelihood cutoff on original alignment: -5403.902\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.987\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 201: -5312.511\n",
      "Iteration 210 / LogL: -5325.149 / Time: 0h:14m:1s (0h:6m:6s left)\n",
      "Iteration 220 / LogL: -5325.220 / Time: 0h:14m:39s (0h:5m:25s left)\n",
      "Iteration 230 / LogL: -5328.189 / Time: 0h:15m:13s (0h:4m:43s left)\n",
      "Iteration 240 / LogL: -5324.069 / Time: 0h:15m:53s (0h:4m:3s left)\n",
      "BETTER TREE FOUND at iteration 242: -5312.511\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 244: -5312.202\n",
      "Iteration 250 / LogL: -5332.035 / Time: 0h:16m:30s (0h:6m:14s left)\n",
      "Log-likelihood cutoff on original alignment: -5403.902\n",
      "Iteration 260 / LogL: -5324.268 / Time: 0h:17m:14s (0h:5m:35s left)\n",
      "Iteration 270 / LogL: -5320.064 / Time: 0h:17m:52s (0h:4m:55s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 278: -5311.706\n",
      "Iteration 280 / LogL: -5314.281 / Time: 0h:18m:30s (0h:6m:29s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 290: -5311.040\n",
      "Iteration 290 / LogL: -5311.040 / Time: 0h:18m:58s (0h:6m:34s left)\n",
      "Iteration 300 / LogL: -5313.877 / Time: 0h:19m:25s (0h:5m:50s left)\n",
      "Log-likelihood cutoff on original alignment: -5403.828\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.986\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 310 / LogL: -5316.220 / Time: 0h:19m:52s (0h:5m:47s left)\n",
      "Iteration 320 / LogL: -5316.283 / Time: 0h:20m:22s (0h:5m:6s left)\n",
      "Iteration 330 / LogL: -5318.158 / Time: 0h:20m:49s (0h:4m:25s left)\n",
      "Iteration 340 / LogL: -5311.041 / Time: 0h:21m:15s (0h:3m:45s left)\n",
      "Iteration 350 / LogL: -5327.419 / Time: 0h:21m:42s (0h:3m:6s left)\n",
      "Log-likelihood cutoff on original alignment: -5403.828\n",
      "Iteration 360 / LogL: -5330.929 / Time: 0h:22m:8s (0h:2m:28s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 368: -5310.677\n",
      "Iteration 370 / LogL: -5319.526 / Time: 0h:22m:39s (0h:6m:0s left)\n",
      "Iteration 380 / LogL: -5330.897 / Time: 0h:23m:4s (0h:5m:21s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 384: -5310.322\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 386: -5309.512\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 388: -5308.339\n",
      "Iteration 390 / LogL: -5346.911 / Time: 0h:23m:34s (0h:5m:56s left)\n",
      "Iteration 400 / LogL: -5312.444 / Time: 0h:24m:2s (0h:5m:18s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.990\n",
      "Iteration 410 / LogL: -5313.044 / Time: 0h:24m:30s (0h:5m:23s left)\n",
      "Iteration 420 / LogL: -5316.978 / Time: 0h:25m:0s (0h:4m:46s left)\n",
      "Iteration 430 / LogL: -5320.180 / Time: 0h:25m:28s (0h:4m:9s left)\n",
      "Iteration 440 / LogL: -5315.757 / Time: 0h:25m:58s (0h:3m:32s left)\n",
      "Iteration 450 / LogL: -5333.857 / Time: 0h:26m:27s (0h:2m:56s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Iteration 460 / LogL: -5319.197 / Time: 0h:26m:53s (0h:2m:20s left)\n",
      "Iteration 470 / LogL: -5343.116 / Time: 0h:27m:23s (0h:1m:45s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 471: -5307.914\n",
      "Iteration 480 / LogL: -5308.589 / Time: 0h:27m:54s (0h:5m:18s left)\n",
      "Iteration 490 / LogL: -5308.696 / Time: 0h:28m:25s (0h:4m:42s left)\n",
      "Iteration 500 / LogL: -5308.757 / Time: 0h:28m:55s (0h:4m:6s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.989\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 501: -5307.067\n",
      "Iteration 510 / LogL: -5315.901 / Time: 0h:29m:25s (0h:5m:15s left)\n",
      "Iteration 520 / LogL: -5311.659 / Time: 0h:29m:52s (0h:4m:39s left)\n",
      "Iteration 530 / LogL: -5315.873 / Time: 0h:30m:18s (0h:4m:4s left)\n",
      "Iteration 540 / LogL: -5309.636 / Time: 0h:30m:47s (0h:3m:29s left)\n",
      "Iteration 550 / LogL: -5313.354 / Time: 0h:31m:16s (0h:2m:54s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 555: -5306.137\n",
      "Iteration 560 / LogL: -5308.121 / Time: 0h:31m:45s (0h:5m:23s left)\n",
      "Iteration 570 / LogL: -5307.679 / Time: 0h:32m:13s (0h:4m:48s left)\n",
      "Iteration 580 / LogL: -5313.962 / Time: 0h:32m:44s (0h:4m:14s left)\n",
      "Iteration 590 / LogL: -5320.208 / Time: 0h:33m:13s (0h:3m:40s left)\n",
      "Iteration 600 / LogL: -5312.530 / Time: 0h:33m:42s (0h:3m:5s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.985\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 610 / LogL: -5317.465 / Time: 0h:34m:11s (0h:5m:3s left)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Iteration 620 / LogL: -5314.305 / Time: 0h:34m:38s (0h:4m:28s left)\n",
      "Iteration 630 / LogL: -5312.157 / Time: 0h:35m:7s (0h:3m:54s left)\n",
      "Iteration 640 / LogL: -5315.054 / Time: 0h:35m:38s (0h:3m:20s left)\n",
      "Iteration 650 / LogL: -5327.857 / Time: 0h:36m:8s (0h:2m:47s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Iteration 660 / LogL: -5307.392 / Time: 0h:36m:40s (0h:2m:13s left)\n",
      "Iteration 670 / LogL: -5329.055 / Time: 0h:37m:8s (0h:1m:39s left)\n",
      "Iteration 680 / LogL: -5306.896 / Time: 0h:37m:33s (0h:1m:6s left)\n",
      "Iteration 690 / LogL: -5316.590 / Time: 0h:38m:3s (0h:0m:33s left)\n",
      "Iteration 700 / LogL: -5319.702 / Time: 0h:38m:29s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.983\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 710 / LogL: -5319.929 / Time: 0h:38m:57s (0h:4m:56s left)\n",
      "Iteration 720 / LogL: -5314.761 / Time: 0h:39m:28s (0h:4m:23s left)\n",
      "Iteration 730 / LogL: -5323.152 / Time: 0h:39m:55s (0h:3m:50s left)\n",
      "Iteration 740 / LogL: -5306.461 / Time: 0h:40m:24s (0h:3m:16s left)\n",
      "Iteration 750 / LogL: -5308.924 / Time: 0h:40m:52s (0h:2m:43s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Iteration 760 / LogL: -5308.552 / Time: 0h:41m:19s (0h:2m:10s left)\n",
      "Iteration 770 / LogL: -5310.675 / Time: 0h:41m:48s (0h:1m:37s left)\n",
      "Iteration 780 / LogL: -5330.090 / Time: 0h:42m:18s (0h:1m:5s left)\n",
      "Iteration 790 / LogL: -5306.934 / Time: 0h:42m:46s (0h:0m:32s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 800: -5306.126\n",
      "Iteration 800 / LogL: -5306.126 / Time: 0h:43m:15s (0h:5m:24s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.986\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 810 / LogL: -5308.409 / Time: 0h:43m:47s (0h:4m:52s left)\n",
      "Iteration 820 / LogL: -5313.900 / Time: 0h:44m:18s (0h:4m:19s left)\n",
      "Iteration 830 / LogL: -5314.211 / Time: 0h:44m:51s (0h:3m:47s left)\n",
      "Iteration 840 / LogL: -5310.284 / Time: 0h:45m:25s (0h:3m:14s left)\n",
      "Iteration 850 / LogL: -5327.535 / Time: 0h:45m:54s (0h:2m:42s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Iteration 860 / LogL: -5314.016 / Time: 0h:46m:26s (0h:2m:9s left)\n",
      "Iteration 870 / LogL: -5313.611 / Time: 0h:46m:56s (0h:1m:37s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 874: -5305.842\n",
      "Iteration 880 / LogL: -5310.232 / Time: 0h:47m:27s (0h:5m:4s left)\n",
      "Iteration 890 / LogL: -5315.809 / Time: 0h:47m:55s (0h:4m:31s left)\n",
      "Iteration 900 / LogL: -5330.178 / Time: 0h:48m:25s (0h:3m:59s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.988\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 910 / LogL: -5307.914 / Time: 0h:48m:55s (0h:4m:50s left)\n",
      "Iteration 920 / LogL: -5314.871 / Time: 0h:49m:27s (0h:4m:18s left)\n",
      "Iteration 930 / LogL: -5317.734 / Time: 0h:50m:1s (0h:3m:46s left)\n",
      "Iteration 940 / LogL: -5309.146 / Time: 0h:50m:34s (0h:3m:13s left)\n",
      "BETTER TREE FOUND at iteration 946: -5305.841\n",
      "Iteration 950 / LogL: -5306.363 / Time: 0h:51m:6s (0h:5m:10s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "Iteration 960 / LogL: -5312.312 / Time: 0h:51m:39s (0h:4m:37s left)\n",
      "Iteration 970 / LogL: -5325.674 / Time: 0h:52m:9s (0h:4m:5s left)\n",
      "Iteration 980 / LogL: -5307.735 / Time: 0h:52m:41s (0h:3m:33s left)\n",
      "Iteration 990 / LogL: -5309.728 / Time: 0h:53m:13s (0h:3m:0s left)\n",
      "Iteration 1000 / LogL: -5305.850 / Time: 0h:53m:42s (0h:2m:28s left)\n",
      "Log-likelihood cutoff on original alignment: -5398.594\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.989\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "WARNING: bootstrap analysis did not converge. You should rerun with higher number of iterations (-nm option)\n",
      "TREE SEARCH COMPLETED AFTER 1001 ITERATIONS / Time: 0h:53m:46s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -5305.841\n",
      "Optimal log-likelihood: -5305.841\n",
      "Gamma shape alpha: 1.348\n",
      "Parameters optimization took 1 rounds (0.140 sec)\n",
      "BEST SCORE FOUND : -5305.841\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 1000 replicates...\n",
      "Testing tree branches by aBayes parametric test...\n",
      "0.574 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file clustalo_strict_trimmed_iqtree_bootstrap.splits.nex\n",
      "Total tree length: 40.701\n",
      "\n",
      "Total number of iterations: 1001\n",
      "CPU time used for tree search: 3208.130 sec (0h:53m:28s)\n",
      "Wall-clock time used for tree search: 3220.406 sec (0h:53m:40s)\n",
      "Total CPU time used: 3215.593 sec (0h:53m:35s)\n",
      "Total wall-clock time used: 3227.870 sec (0h:53m:47s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file clustalo_strict_trimmed_iqtree_bootstrap.splits.nex...\n",
      "196 taxa and 2463 splits.\n",
      "Consensus tree written to clustalo_strict_trimmed_iqtree_bootstrap.contree\n",
      "Reading input trees file clustalo_strict_trimmed_iqtree_bootstrap.contree\n",
      "Log-likelihood of consensus tree: -5307.118\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                clustalo_strict_trimmed_iqtree_bootstrap.iqtree\n",
      "  Maximum-likelihood tree:       clustalo_strict_trimmed_iqtree_bootstrap.treefile\n",
      "  Likelihood distances:          clustalo_strict_trimmed_iqtree_bootstrap.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          clustalo_strict_trimmed_iqtree_bootstrap.splits.nex\n",
      "  Consensus tree:                clustalo_strict_trimmed_iqtree_bootstrap.contree\n",
      "  Screen log file:               clustalo_strict_trimmed_iqtree_bootstrap.log\n",
      "\n",
      "Date and Time: Sat May 21 13:57:35 2022\n"
     ]
    }
   ],
   "source": [
    "!iqtree -m MFP -s strict_trimmed_alig_clustalo.fa --prefix clustalo_strict_trimmed_iqtree_bootstrap -o ERN08734_Amborella_trichopoda -B 1000 -alrt 1000 -abayes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa3fc8e0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
